{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "from selenium import webdriver\n",
    "from selenium.webdriver.common.by import By\n",
    "from selenium.webdriver.common.keys import Keys\n",
    "from selenium.webdriver.support.ui import Select\n",
    "from selenium.webdriver.support import expected_conditions as EC\n",
    "from selenium.webdriver.support.ui import WebDriverWait as wait\n",
    "\n",
    "import pandas as pd\n",
    "\n",
    "import time\n",
    "import datetime\n",
    "import threading\n",
    "\n",
    "from bs4 import BeautifulSoup as bs\n",
    "\n",
    "import smtplib\n",
    "from email.mime.multipart import MIMEMultipart"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "105 2019/07/26\n",
      "page not found\n",
      "106 2019/07/27\n",
      "page not found\n",
      "107 2019/07/28\n",
      "page not found\n",
      "108 2019/07/29\n",
      "page not found\n",
      "109 2019/07/30\n",
      "page not found\n",
      "110 2019/07/31\n",
      "page not found\n",
      "111 2019/08/01\n",
      "page not found\n",
      "112 2019/08/02\n",
      "page not found\n",
      "113 2019/08/03\n",
      "page not found\n",
      "114 2019/08/04\n",
      "page not found\n",
      "115 2019/08/05\n",
      "page not found\n",
      "116 2019/08/06\n",
      "page not found\n",
      "117 2019/08/07\n",
      "page not found\n",
      "118 2019/08/08\n",
      "page not found\n",
      "119 2019/08/09\n",
      "page not found\n",
      "120 2019/08/10\n",
      "page not found\n",
      "121 2019/08/11\n",
      "page not found\n",
      "122 2019/08/12\n",
      "page not found\n",
      "123 2019/08/13\n",
      "page not found\n",
      "124 2019/08/14\n",
      "Message: Cannot locate option with value: XXX\n",
      "\n",
      "125 2019/08/15\n",
      "page not found\n",
      "126 2019/08/16\n",
      "page not found\n",
      "127 2019/08/17\n",
      "page not found\n",
      "128 2019/08/18\n",
      "page not found\n",
      "129 2019/08/19\n",
      "page not found\n",
      "130 2019/08/20\n",
      "page not found\n",
      "131 2019/08/21\n",
      "page not found\n",
      "132 2019/08/22\n",
      "page not found\n",
      "133 2019/08/23\n",
      "page not found\n",
      "134 2019/08/24\n",
      "page not found\n",
      "135 2019/08/25\n",
      "page not found\n",
      "136 2019/08/26\n",
      "page not found\n",
      "137 2019/08/27\n",
      "page not found\n",
      "138 2019/08/28\n",
      "page not found\n",
      "139 2019/08/29\n",
      "page not found\n",
      "140 2019/08/30\n",
      "page not found\n",
      "141 2019/08/31\n",
      "page not found\n",
      "142 2019/09/01\n",
      "page not found\n",
      "143 2019/09/02\n",
      "page not found\n",
      "144 2019/09/03\n",
      "page not found\n",
      "145 2019/09/04\n",
      "page not found\n",
      "146 2019/09/05\n",
      "page not found\n",
      "147 2019/09/06\n",
      "page not found\n",
      "148 2019/09/07\n",
      "page not found\n",
      "149 2019/09/08\n",
      "page not found\n",
      "150 2019/09/09\n",
      "page not found\n",
      "151 2019/09/10\n",
      "page not found\n",
      "152 2019/09/11\n",
      "page not found\n",
      "153 2019/09/12\n",
      "page not found\n",
      "154 2019/09/13\n",
      "page not found\n",
      "155 2019/09/14\n",
      "page not found\n",
      "156 2019/09/15\n",
      "page not found\n",
      "157 2019/09/16\n",
      "page not found\n",
      "158 2019/09/17\n",
      "page not found\n",
      "159 2019/09/18\n",
      "page not found\n",
      "160 2019/09/19\n",
      "page not found\n",
      "161 2019/09/20\n",
      "page not found\n",
      "162 2019/09/21\n",
      "page not found\n",
      "163 2019/09/22\n",
      "page not found\n",
      "164 2019/09/23\n",
      "page not found\n",
      "165 2019/09/24\n",
      "page not found\n",
      "166 2019/09/25\n",
      "page not found\n",
      "167 2019/09/26\n",
      "page not found\n",
      "168 2019/09/27\n",
      "page not found\n",
      "169 2019/09/28\n",
      "page not found\n",
      "170 2019/09/29\n",
      "page not found\n",
      "171 2019/09/30\n",
      "page not found\n",
      "172 2019/10/01\n",
      "page not found\n",
      "173 2019/10/02\n",
      "page not found\n",
      "174 2019/10/03\n",
      "page not found\n",
      "175 2019/10/04\n",
      "page not found\n",
      "176 2019/10/05\n",
      "page not found\n",
      "177 2019/10/06\n",
      "page not found\n",
      "178 2019/10/07\n",
      "page not found\n",
      "179 2019/10/08\n",
      "page not found\n",
      "180 2019/10/09\n",
      "page not found\n",
      "181 2019/10/10\n",
      "page not found\n",
      "182 2019/10/11\n",
      "page not found\n",
      "183 2019/10/12\n",
      "page not found\n",
      "184 2019/10/13\n",
      "page not found\n",
      "185 2019/10/14\n",
      "page not found\n",
      "186 2019/10/15\n",
      "page not found\n",
      "187 2019/10/16\n",
      "page not found\n",
      "188 2019/10/17\n",
      "page not found\n",
      "189 2019/10/18\n",
      "page not found\n",
      "190 2019/10/19\n",
      "page not found\n",
      "191 2019/10/20\n",
      "page not found\n",
      "192 2019/10/21\n",
      "page not found\n",
      "193 2019/10/22\n",
      "page not found\n",
      "194 2019/10/23\n",
      "page not found\n",
      "195 2019/10/24\n",
      "page not found\n",
      "196 2019/10/25\n",
      "page not found\n",
      "197 2019/10/26\n",
      "page not found\n",
      "198 2019/10/27\n",
      "page not found\n",
      "199 2019/10/28\n",
      "page not found\n",
      "200 2019/10/29\n",
      "page not found\n",
      "201 2019/10/30\n",
      "page not found\n",
      "202 2019/10/31\n",
      "page not found\n",
      "203 2019/11/01\n",
      "page not found\n",
      "204 2019/11/02\n",
      "Message: unknown error: Cannot read property 'setAttribute' of undefined\n",
      "  (Session info: chrome=73.0.3683.103)\n",
      "  (Driver info: chromedriver=2.45.615291 (ec3682e3c9061c10f26ea9e5cdcf3c53f3f74387),platform=Windows NT 10.0.17134 x86_64)\n",
      "\n",
      "205 2019/11/03\n",
      "page not found\n",
      "206 2019/11/04\n",
      "page not found\n",
      "207 2019/11/05\n",
      "page not found\n",
      "208 2019/11/06\n",
      "page not found\n",
      "209 2019/11/07\n",
      "page not found\n",
      "210 2019/11/08\n",
      "page not found\n",
      "211 2019/11/09\n",
      "page not found\n",
      "212 2019/11/10\n",
      "page not found\n",
      "213 2019/11/11\n",
      "page not found\n",
      "214 2019/11/12\n",
      "page not found\n",
      "215 2019/11/13\n",
      "page not found\n",
      "216 2019/11/14\n",
      "page not found\n",
      "217 2019/11/15\n",
      "page not found\n",
      "218 2019/11/16\n",
      "page not found\n",
      "219 2019/11/17\n",
      "page not found\n",
      "220 2019/11/18\n",
      "Message: Cannot locate option with value: XXX\n",
      "\n",
      "221 2019/11/19\n",
      "page not found\n",
      "222 2019/11/20\n",
      "page not found\n",
      "223 2019/11/21\n",
      "page not found\n",
      "224 2019/11/22\n",
      "page not found\n",
      "225 2019/11/23\n",
      "page not found\n",
      "226 2019/11/24\n",
      "page not found\n",
      "227 2019/11/25\n",
      "page not found\n",
      "228 2019/11/26\n",
      "page not found\n",
      "229 2019/11/27\n",
      "page not found\n",
      "230 2019/11/28\n",
      "page not found\n",
      "231 2019/11/29\n",
      "page not found\n",
      "232 2019/11/30\n",
      "page not found\n",
      "233 2019/12/01\n",
      "page not found\n",
      "234 2019/12/02\n",
      "page not found\n",
      "235 2019/12/03\n",
      "page not found\n",
      "236 2019/12/04\n",
      "page not found\n",
      "237 2019/12/05\n",
      "page not found\n",
      "238 2019/12/06\n",
      "page not found\n",
      "239 2019/12/07\n",
      "page not found\n",
      "240 2019/12/08\n",
      "page not found\n",
      "241 2019/12/09\n",
      "page not found\n",
      "242 2019/12/10\n",
      "page not found\n",
      "243 2019/12/11\n",
      "page not found\n",
      "244 2019/12/12\n",
      "page not found\n",
      "245 2019/12/13\n",
      "page not found\n",
      "246 2019/12/14\n",
      "page not found\n",
      "247 2019/12/15\n",
      "page not found\n",
      "248 2019/12/16\n",
      "page not found\n",
      "249 2019/12/17\n",
      "page not found\n",
      "250 2019/12/18\n",
      "page not found\n",
      "251 2019/12/19\n",
      "page not found\n",
      "252 2019/12/20\n",
      "page not found\n",
      "253 2019/12/21\n",
      "page not found\n",
      "254 2019/12/22\n",
      "page not found\n",
      "255 2019/12/23\n",
      "page not found\n",
      "256 2019/12/24\n",
      "page not found\n",
      "257 2019/12/25\n",
      "page not found\n",
      "258 2019/12/26\n",
      "page not found\n",
      "259 2019/12/27\n",
      "page not found\n",
      "260 2019/12/28\n",
      "page not found\n",
      "261 2019/12/29\n",
      "page not found\n",
      "262 2019/12/30\n",
      "page not found\n",
      "263 2019/12/31\n",
      "page not found\n",
      "264 2020/01/01\n",
      "page not found\n",
      "265 2020/01/02\n",
      "page not found\n",
      "266 2020/01/03\n",
      "page not found\n",
      "267 2020/01/04\n",
      "page not found\n",
      "268 2020/01/05\n",
      "page not found\n",
      "269 2020/01/06\n",
      "page not found\n",
      "270 2020/01/07\n",
      "page not found\n",
      "271 2020/01/08\n",
      "page not found\n",
      "272 2020/01/09\n",
      "page not found\n",
      "273 2020/01/10\n",
      "page not found\n",
      "274 2020/01/11\n",
      "page not found\n",
      "275 2020/01/12\n",
      "page not found\n",
      "276 2020/01/13\n",
      "page not found\n",
      "277 2020/01/14\n",
      "page not found\n",
      "278 2020/01/15\n",
      "page not found\n",
      "279 2020/01/16\n",
      "Message: Cannot locate option with value: XXX\n",
      "\n",
      "280 2020/01/17\n",
      "page not found\n",
      "281 2020/01/18\n",
      "page not found\n",
      "282 2020/01/19\n",
      "page not found\n",
      "283 2020/01/20\n",
      "page not found\n",
      "284 2020/01/21\n",
      "page not found\n",
      "285 2020/01/22\n",
      "page not found\n",
      "286 2020/01/23\n",
      "page not found\n",
      "287 2020/01/24\n",
      "page not found\n",
      "288 2020/01/25\n",
      "page not found\n",
      "289 2020/01/26\n",
      "page not found\n",
      "290 2020/01/27\n",
      "page not found\n",
      "291 2020/01/28\n",
      "page not found\n",
      "292 2020/01/29\n",
      "page not found\n",
      "293 2020/01/30\n",
      "page not found\n",
      "294 2020/01/31\n",
      "page not found\n",
      "295 2020/02/01\n",
      "page not found\n",
      "296 2020/02/02\n",
      "page not found\n",
      "297 2020/02/03\n",
      "page not found\n",
      "298 2020/02/04\n",
      "page not found\n",
      "299 2020/02/05\n",
      "page not found\n"
     ]
    }
   ],
   "source": [
    "flag = 0\n",
    "for i in range(105,300):\n",
    "    today = datetime.datetime.now() + datetime.timedelta(days=i)\n",
    "    tostr = today.strftime('%Y/%m/%d')\n",
    "    #     end_date = today + datetime.timedelta(days=5) + datetime.timedelta(days=i)\n",
    "    #     enstr = end_date.strftime('%Y/%m/%d')\n",
    "\n",
    "    browser = webdriver.Chrome(executable_path='driver/chromedriver')\n",
    "    browser.get('https://www.tigerairtw.com/zh-tw')\n",
    "    print(i,tostr)\n",
    "    #     print(tostr,enstr)\n",
    "\n",
    "    try:\n",
    "        time.sleep(4)\n",
    "        #checkbox\n",
    "        browser.execute_script(\"document.getElementsByClassName('class_type')[1].setAttribute('checked', true)\")\n",
    "    #         checkboxs=browser.find_elements_by_class_name('class_type')\n",
    "    #         for checkbox in checkboxs:\n",
    "    #             print(checkbox.get_attribute('value'),checkbox.get_attribute('checked')\n",
    "\n",
    "        # select part\n",
    "        s1 = Select(browser.find_element_by_class_name('selectOrigin'))\n",
    "    #     for select in s1.options:\n",
    "    #         print(select.text, select.get_attribute('value'))\n",
    "        s1.select_by_value(\"XX3\")\n",
    "        s2 = Select(browser.find_element_by_class_name('selectDestination'))\n",
    "        # for select in s2.options:\n",
    "        #     print(select.text)\n",
    "        s2.select_by_value('XXX')\n",
    "\n",
    "        # The data table\n",
    "        browser.execute_script(\"var t1 = document.getElementsByClassName('datepickerfrom')[0]; t1.value='\" +str(tostr)+ \"';\");\n",
    "    #         browser.execute_script(\"var t2 = document.getElementsByClassName('datepickerto')[0]; t2.value='\" +str(enstr)+ \"';\");\n",
    "\n",
    "        #how many people\n",
    "    #     browser.execute_script(\"var t3 = document.getElementsByClassName('total_pass_input')[0]; t3.value=3;\");\n",
    "    #     browser.execute_script(\"var t2 = document.getElementsByClassName('datepickerto')[0]; t2.value='2019/04/17';\");\n",
    "\n",
    "\n",
    "        browser.find_element_by_css_selector('.btn.btn-primary').submit()\n",
    "        time.sleep(10)\n",
    "\n",
    "        try:\n",
    "            if browser.find_element_by_class_name('error-title') != []:\n",
    "                print(browser.find_element_by_class_name('error-title').text)\n",
    "        except:\n",
    "            time.sleep(10)\n",
    "            print('no error')\n",
    "            html = browser.page_source\n",
    "            soup = bs(html)\n",
    "            flag = 1\n",
    "            pass\n",
    "        if flag == 1:\n",
    "            break\n",
    "\n",
    "    except Exception as e:\n",
    "        print(str(e))\n",
    "        pass\n",
    "\n",
    "    browser.quit()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "0 2019/04/12\n",
      "page not found\n",
      "1 2019/04/13\n",
      "page not found\n",
      "2 2019/04/14\n",
      "page not found\n",
      "3 2019/04/15\n",
      "page not found\n",
      "4 2019/04/16\n",
      "page not found\n",
      "5 2019/04/17\n",
      "page not found\n",
      "6 2019/04/18\n",
      "page not found\n",
      "7 2019/04/19\n",
      "page not found\n",
      "8 2019/04/20\n",
      "page not found\n",
      "9 2019/04/21\n",
      "page not found\n",
      "10 2019/04/22\n",
      "page not found\n",
      "11 2019/04/23\n",
      "page not found\n",
      "12 2019/04/24\n",
      "page not found\n",
      "13 2019/04/25\n",
      "page not found\n",
      "14 2019/04/26\n",
      "page not found\n",
      "15 2019/04/27\n",
      "page not found\n",
      "16 2019/04/28\n",
      "page not found\n",
      "17 2019/04/29\n",
      "page not found\n",
      "18 2019/04/30\n",
      "page not found\n",
      "19 2019/05/01\n",
      "page not found\n",
      "20 2019/05/02\n",
      "page not found\n",
      "21 2019/05/03\n",
      "page not found\n",
      "22 2019/05/04\n",
      "page not found\n",
      "23 2019/05/05\n",
      "page not found\n",
      "24 2019/05/06\n",
      "page not found\n",
      "25 2019/05/07\n",
      "page not found\n",
      "26 2019/05/08\n",
      "page not found\n",
      "27 2019/05/09\n",
      "page not found\n",
      "28 2019/05/10\n",
      "page not found\n",
      "29 2019/05/11\n",
      "page not found\n",
      "30 2019/05/12\n",
      "page not found\n",
      "31 2019/05/13\n",
      "page not found\n",
      "32 2019/05/14\n",
      "page not found\n",
      "33 2019/05/15\n",
      "page not found\n",
      "34 2019/05/16\n",
      "page not found\n",
      "35 2019/05/17\n",
      "page not found\n",
      "36 2019/05/18\n",
      "page not found\n",
      "37 2019/05/19\n",
      "page not found\n",
      "38 2019/05/20\n",
      "page not found\n",
      "39 2019/05/21\n",
      "page not found\n",
      "40 2019/05/22\n",
      "page not found\n",
      "41 2019/05/23\n",
      "page not found\n",
      "42 2019/05/24\n",
      "page not found\n",
      "43 2019/05/25\n",
      "page not found\n",
      "44 2019/05/26\n",
      "page not found\n",
      "45 2019/05/27\n",
      "page not found\n",
      "46 2019/05/28\n",
      "page not found\n",
      "47 2019/05/29\n",
      "page not found\n",
      "48 2019/05/30\n",
      "page not found\n",
      "49 2019/05/31\n",
      "page not found\n",
      "50 2019/06/01\n",
      "page not found\n",
      "51 2019/06/02\n",
      "page not found\n",
      "52 2019/06/03\n",
      "page not found\n",
      "53 2019/06/04\n",
      "page not found\n",
      "54 2019/06/05\n",
      "page not found\n",
      "55 2019/06/06\n",
      "page not found\n",
      "56 2019/06/07\n",
      "page not found\n",
      "57 2019/06/08\n",
      "page not found\n",
      "58 2019/06/09\n",
      "page not found\n",
      "59 2019/06/10\n",
      "page not found\n",
      "60 2019/06/11\n",
      "page not found\n",
      "61 2019/06/12\n",
      "page not found\n",
      "62 2019/06/13\n",
      "page not found\n",
      "63 2019/06/14\n",
      "page not found\n",
      "64 2019/06/15\n",
      "page not found\n",
      "65 2019/06/16\n",
      "page not found\n",
      "66 2019/06/17\n",
      "page not found\n",
      "67 2019/06/18\n",
      "page not found\n",
      "68 2019/06/19\n",
      "page not found\n",
      "69 2019/06/20\n",
      "page not found\n",
      "70 2019/06/21\n",
      "page not found\n",
      "71 2019/06/22\n",
      "page not found\n",
      "72 2019/06/23\n",
      "page not found\n",
      "73 2019/06/24\n",
      "page not found\n",
      "74 2019/06/25\n",
      "page not found\n",
      "75 2019/06/26\n",
      "page not found\n",
      "76 2019/06/27\n",
      "page not found\n",
      "77 2019/06/28\n",
      "page not found\n",
      "78 2019/06/29\n",
      "page not found\n",
      "79 2019/06/30\n",
      "page not found\n",
      "80 2019/07/01\n",
      "page not found\n",
      "81 2019/07/02\n",
      "page not found\n",
      "82 2019/07/03\n",
      "page not found\n",
      "83 2019/07/04\n",
      "page not found\n",
      "84 2019/07/05\n",
      "page not found\n",
      "85 2019/07/06\n",
      "page not found\n",
      "86 2019/07/07\n",
      "page not found\n",
      "87 2019/07/08\n",
      "page not found\n",
      "88 2019/07/09\n",
      "page not found\n",
      "89 2019/07/10\n",
      "page not found\n",
      "90 2019/07/11\n",
      "page not found\n",
      "91 2019/07/12\n",
      "page not found\n",
      "92 2019/07/13\n",
      "page not found\n",
      "93 2019/07/14\n",
      "page not found\n",
      "94 2019/07/15\n",
      "page not found\n",
      "95 2019/07/16\n",
      "page not found\n",
      "96 2019/07/17\n",
      "page not found\n",
      "97 2019/07/18\n",
      "page not found\n",
      "98 2019/07/19\n",
      "page not found\n",
      "99 2019/07/20\n",
      "page not found\n",
      "100 2019/07/21\n",
      "page not found\n",
      "101 2019/07/22\n",
      "page not found\n",
      "102 2019/07/23\n",
      "page not found\n",
      "103 2019/07/24\n",
      "page not found\n",
      "104 2019/07/25\n",
      "page not found\n",
      "105 2019/07/26\n",
      "page not found\n",
      "106 2019/07/27\n",
      "Message: Cannot locate option with value: XX4\n",
      "\n",
      "107 2019/07/28\n",
      "page not found\n",
      "108 2019/07/29\n",
      "page not found\n",
      "109 2019/07/30\n",
      "page not found\n",
      "110 2019/07/31\n",
      "page not found\n",
      "111 2019/08/01\n",
      "page not found\n",
      "112 2019/08/02\n",
      "page not found\n",
      "113 2019/08/03\n",
      "page not found\n",
      "114 2019/08/04\n",
      "page not found\n",
      "115 2019/08/05\n",
      "page not found\n",
      "116 2019/08/06\n",
      "page not found\n",
      "117 2019/08/07\n",
      "page not found\n",
      "118 2019/08/08\n",
      "page not found\n",
      "119 2019/08/09\n",
      "page not found\n",
      "120 2019/08/10\n",
      "page not found\n",
      "121 2019/08/11\n",
      "page not found\n",
      "122 2019/08/12\n",
      "page not found\n",
      "123 2019/08/13\n",
      "page not found\n",
      "124 2019/08/15\n",
      "page not found\n",
      "125 2019/08/16\n",
      "page not found\n",
      "126 2019/08/17\n",
      "page not found\n",
      "127 2019/08/18\n",
      "page not found\n",
      "128 2019/08/19\n",
      "page not found\n",
      "129 2019/08/20\n",
      "page not found\n",
      "130 2019/08/21\n",
      "page not found\n",
      "131 2019/08/22\n",
      "page not found\n",
      "132 2019/08/23\n",
      "page not found\n",
      "133 2019/08/24\n",
      "page not found\n",
      "134 2019/08/25\n",
      "Message: Cannot locate option with value: XX4\n",
      "\n",
      "135 2019/08/26\n",
      "page not found\n",
      "136 2019/08/27\n",
      "page not found\n",
      "137 2019/08/28\n",
      "page not found\n",
      "138 2019/08/29\n",
      "page not found\n",
      "139 2019/08/30\n",
      "page not found\n",
      "140 2019/08/31\n",
      "page not found\n",
      "141 2019/09/01\n",
      "page not found\n",
      "142 2019/09/02\n",
      "page not found\n",
      "143 2019/09/03\n",
      "page not found\n",
      "144 2019/09/04\n",
      "page not found\n",
      "145 2019/09/05\n",
      "page not found\n",
      "146 2019/09/06\n",
      "page not found\n",
      "147 2019/09/07\n",
      "page not found\n",
      "148 2019/09/08\n",
      "page not found\n",
      "149 2019/09/09\n",
      "page not found\n",
      "150 2019/09/10\n",
      "page not found\n",
      "151 2019/09/11\n",
      "page not found\n",
      "152 2019/09/12\n",
      "page not found\n",
      "153 2019/09/13\n",
      "page not found\n",
      "154 2019/09/14\n",
      "page not found\n",
      "155 2019/09/15\n",
      "page not found\n",
      "156 2019/09/16\n",
      "page not found\n",
      "157 2019/09/17\n",
      "page not found\n",
      "158 2019/09/18\n",
      "page not found\n",
      "159 2019/09/19\n",
      "page not found\n",
      "160 2019/09/20\n",
      "page not found\n",
      "161 2019/09/21\n",
      "page not found\n",
      "162 2019/09/22\n",
      "page not found\n",
      "163 2019/09/23\n",
      "page not found\n",
      "164 2019/09/24\n",
      "page not found\n",
      "165 2019/09/25\n",
      "page not found\n",
      "166 2019/09/26\n",
      "page not found\n",
      "167 2019/09/27\n",
      "page not found\n",
      "168 2019/09/28\n",
      "page not found\n",
      "169 2019/09/29\n",
      "page not found\n",
      "170 2019/09/30\n",
      "page not found\n",
      "171 2019/10/01\n",
      "page not found\n",
      "172 2019/10/02\n",
      "page not found\n",
      "173 2019/10/03\n",
      "page not found\n",
      "174 2019/10/04\n",
      "page not found\n",
      "175 2019/10/05\n",
      "page not found\n",
      "176 2019/10/06\n",
      "page not found\n",
      "177 2019/10/07\n",
      "page not found\n",
      "178 2019/10/08\n",
      "page not found\n",
      "179 2019/10/09\n",
      "page not found\n",
      "180 2019/10/10\n",
      "page not found\n",
      "181 2019/10/11\n",
      "page not found\n",
      "182 2019/10/12\n",
      "page not found\n",
      "183 2019/10/13\n",
      "page not found\n",
      "184 2019/10/14\n",
      "page not found\n",
      "185 2019/10/15\n",
      "page not found\n",
      "186 2019/10/16\n",
      "page not found\n",
      "187 2019/10/17\n",
      "page not found\n",
      "188 2019/10/18\n",
      "page not found\n",
      "189 2019/10/19\n",
      "page not found\n",
      "190 2019/10/20\n",
      "page not found\n",
      "191 2019/10/21\n",
      "page not found\n",
      "192 2019/10/22\n",
      "page not found\n",
      "193 2019/10/23\n",
      "page not found\n",
      "194 2019/10/24\n",
      "page not found\n",
      "195 2019/10/25\n",
      "page not found\n",
      "196 2019/10/26\n",
      "page not found\n",
      "197 2019/10/27\n",
      "page not found\n",
      "198 2019/10/28\n",
      "page not found\n",
      "199 2019/10/29\n",
      "page not found\n",
      "200 2019/10/30\n",
      "page not found\n",
      "201 2019/10/31\n",
      "page not found\n",
      "202 2019/11/01\n",
      "page not found\n",
      "203 2019/11/02\n",
      "page not found\n",
      "204 2019/11/03\n",
      "page not found\n",
      "205 2019/11/04\n",
      "page not found\n",
      "206 2019/11/05\n",
      "page not found\n",
      "207 2019/11/06\n",
      "page not found\n",
      "208 2019/11/07\n",
      "page not found\n",
      "209 2019/11/08\n",
      "page not found\n",
      "210 2019/11/09\n",
      "page not found\n",
      "211 2019/11/10\n",
      "page not found\n",
      "212 2019/11/11\n",
      "page not found\n",
      "213 2019/11/12\n",
      "page not found\n",
      "214 2019/11/13\n",
      "page not found\n",
      "215 2019/11/14\n",
      "page not found\n",
      "216 2019/11/15\n",
      "page not found\n",
      "217 2019/11/16\n",
      "page not found\n",
      "218 2019/11/17\n",
      "page not found\n",
      "219 2019/11/18\n",
      "page not found\n",
      "220 2019/11/19\n",
      "page not found\n",
      "221 2019/11/20\n",
      "page not found\n",
      "222 2019/11/21\n",
      "page not found\n",
      "223 2019/11/22\n",
      "page not found\n",
      "224 2019/11/23\n",
      "page not found\n",
      "225 2019/11/24\n",
      "page not found\n",
      "226 2019/11/25\n",
      "page not found\n",
      "227 2019/11/26\n",
      "page not found\n",
      "228 2019/11/27\n",
      "page not found\n",
      "229 2019/11/28\n",
      "page not found\n",
      "230 2019/11/29\n",
      "page not found\n",
      "231 2019/11/30\n",
      "page not found\n",
      "232 2019/12/01\n",
      "page not found\n",
      "233 2019/12/02\n",
      "page not found\n",
      "234 2019/12/03\n",
      "page not found\n",
      "235 2019/12/04\n",
      "page not found\n",
      "236 2019/12/05\n",
      "page not found\n",
      "237 2019/12/06\n",
      "page not found\n",
      "238 2019/12/07\n",
      "page not found\n",
      "239 2019/12/08\n",
      "page not found\n",
      "240 2019/12/09\n",
      "page not found\n",
      "241 2019/12/10\n",
      "page not found\n",
      "242 2019/12/11\n",
      "page not found\n",
      "243 2019/12/12\n",
      "page not found\n",
      "244 2019/12/13\n",
      "page not found\n",
      "245 2019/12/14\n",
      "page not found\n",
      "246 2019/12/15\n",
      "page not found\n",
      "247 2019/12/16\n",
      "page not found\n",
      "248 2019/12/17\n",
      "page not found\n",
      "249 2019/12/18\n",
      "page not found\n",
      "250 2019/12/19\n",
      "page not found\n",
      "251 2019/12/20\n",
      "page not found\n",
      "252 2019/12/21\n",
      "page not found\n",
      "253 2019/12/22\n",
      "page not found\n",
      "254 2019/12/23\n",
      "page not found\n",
      "255 2019/12/24\n",
      "page not found\n",
      "256 2019/12/25\n",
      "page not found\n",
      "257 2019/12/26\n",
      "page not found\n",
      "258 2019/12/27\n",
      "page not found\n",
      "259 2019/12/28\n",
      "page not found\n",
      "260 2019/12/29\n",
      "page not found\n",
      "261 2019/12/30\n",
      "page not found\n",
      "262 2019/12/31\n",
      "page not found\n",
      "263 2020/01/01\n",
      "page not found\n",
      "264 2020/01/02\n",
      "page not found\n",
      "265 2020/01/03\n",
      "page not found\n",
      "266 2020/01/04\n",
      "page not found\n",
      "267 2020/01/05\n",
      "page not found\n",
      "268 2020/01/06\n",
      "page not found\n",
      "269 2020/01/07\n",
      "page not found\n",
      "270 2020/01/08\n",
      "page not found\n",
      "271 2020/01/09\n",
      "page not found\n",
      "272 2020/01/10\n",
      "page not found\n",
      "273 2020/01/11\n",
      "page not found\n",
      "274 2020/01/12\n",
      "page not found\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "275 2020/01/13\n",
      "page not found\n",
      "276 2020/01/14\n",
      "page not found\n",
      "277 2020/01/15\n",
      "page not found\n",
      "278 2020/01/16\n",
      "page not found\n",
      "279 2020/01/17\n",
      "page not found\n",
      "280 2020/01/18\n",
      "page not found\n",
      "281 2020/01/19\n",
      "page not found\n",
      "282 2020/01/20\n",
      "page not found\n",
      "283 2020/01/21\n",
      "page not found\n",
      "284 2020/01/22\n",
      "page not found\n",
      "285 2020/01/23\n",
      "page not found\n",
      "286 2020/01/24\n",
      "page not found\n",
      "287 2020/01/25\n",
      "page not found\n",
      "288 2020/01/26\n",
      "page not found\n",
      "289 2020/01/27\n",
      "page not found\n",
      "290 2020/01/28\n",
      "page not found\n",
      "291 2020/01/29\n",
      "page not found\n",
      "292 2020/01/30\n",
      "page not found\n",
      "293 2020/01/31\n",
      "page not found\n",
      "294 2020/02/01\n",
      "page not found\n",
      "295 2020/02/02\n",
      "page not found\n",
      "296 2020/02/03\n",
      "page not found\n",
      "297 2020/02/04\n",
      "page not found\n",
      "298 2020/02/05\n",
      "page not found\n",
      "299 2020/02/06\n",
      "page not found\n"
     ]
    }
   ],
   "source": [
    "flag = 0\n",
    "for i in range(300):\n",
    "    today = datetime.datetime.now() + datetime.timedelta(days=i)\n",
    "    tostr = today.strftime('%Y/%m/%d')\n",
    "    #     end_date = today + datetime.timedelta(days=5) + datetime.timedelta(days=i)\n",
    "    #     enstr = end_date.strftime('%Y/%m/%d')\n",
    "\n",
    "    browser = webdriver.Chrome(executable_path='driver/chromedriver')\n",
    "    browser.get('https://www.tigerairt-w.com/zh-tw')\n",
    "    print(i,tostr)\n",
    "    #     print(tostr,enstr)\n",
    "\n",
    "    try:\n",
    "        time.sleep(4)\n",
    "        #checkbox\n",
    "        browser.execute_script(\"document.getElementsByClassName('class_type')[1].setAttribute('checked', true)\")\n",
    "    #         checkboxs=browser.find_elements_by_class_name('class_type')\n",
    "    #         for checkbox in checkboxs:\n",
    "    #             print(checkbox.get_attribute('value'),checkbox.get_attribute('checked')\n",
    "\n",
    "        # select part\n",
    "        s1 = Select(browser.find_element_by_class_name('selectOrigin'))\n",
    "    #     for select in s1.options:\n",
    "    #         print(select.text, select.get_attribute('value'))\n",
    "        s1.select_by_value(\"XX3\")\n",
    "        s2 = Select(browser.find_element_by_class_name('selectDestination'))\n",
    "        # for select in s2.options:\n",
    "        #     print(select.text)\n",
    "        s2.select_by_value('XX4')\n",
    "\n",
    "        # The data table\n",
    "        browser.execute_script(\"var t1 = document.getElementsByClassName('datepickerfrom')[0]; t1.value='\" +str(tostr)+ \"';\");\n",
    "    #         browser.execute_script(\"var t2 = document.getElementsByClassName('datepickerto')[0]; t2.value='\" +str(enstr)+ \"';\");\n",
    "\n",
    "        #how many people\n",
    "    #     browser.execute_script(\"var t3 = document.getElementsByClassName('total_pass_input')[0]; t3.value=3;\");\n",
    "    #     browser.execute_script(\"var t2 = document.getElementsByClassName('datepickerto')[0]; t2.value='2019/04/17';\");\n",
    "\n",
    "\n",
    "        browser.find_element_by_css_selector('.btn.btn-primary').submit()\n",
    "        time.sleep(10)\n",
    "\n",
    "        try:\n",
    "            if browser.find_element_by_class_name('error-title') != []:\n",
    "                print(browser.find_element_by_class_name('error-title').text)\n",
    "        except:\n",
    "            time.sleep(10)\n",
    "            print('no error')\n",
    "            html = browser.page_source\n",
    "            soup = bs(html)\n",
    "            flag = 1\n",
    "            pass\n",
    "        if flag == 1:\n",
    "            break\n",
    "\n",
    "    except Exception as e:\n",
    "        print(str(e))\n",
    "        pass\n",
    "\n",
    "    browser.quit()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2019/06/24\n",
      "page not found\n"
     ]
    }
   ],
   "source": [
    "today = datetime.datetime.now() + datetime.timedelta(days=72)\n",
    "tostr = today.strftime('%Y/%m/%d')\n",
    "#     end_date = today + datetime.timedelta(days=5) + datetime.timedelta(days=i)\n",
    "#     enstr = end_date.strftime('%Y/%m/%d')\n",
    "\n",
    "browser = webdriver.Chrome(executable_path='driver/chromedriver')\n",
    "browser.get('https://www.tigerairtw.com/zh-tw')\n",
    "print(tostr)\n",
    "#     print(tostr,enstr)\n",
    "\n",
    "try:\n",
    "    time.sleep(4)\n",
    "    #checkbox\n",
    "    browser.execute_script(\"document.getElementsByClassName('class_type')[1].setAttribute('checked', true)\")\n",
    "#         checkboxs=browser.find_elements_by_class_name('class_type')\n",
    "#         for checkbox in checkboxs:\n",
    "#             print(checkbox.get_attribute('value'),checkbox.get_attribute('checked')\n",
    "\n",
    "    # select part\n",
    "    s1 = Select(browser.find_element_by_class_name('selectOrigin'))\n",
    "#     for select in s1.options:\n",
    "#         print(select.text, select.get_attribute('value'))\n",
    "    s1.select_by_value(\"XX3\")\n",
    "    s2 = Select(browser.find_element_by_class_name('selectDestination'))\n",
    "    # for select in s2.options:\n",
    "    #     print(select.text)\n",
    "    s2.select_by_value('XX8')\n",
    "\n",
    "    # The data table\n",
    "    browser.execute_script(\"var t1 = document.getElementsByClassName('datepickerfrom')[0]; t1.value='\" +str(tostr)+ \"';\");\n",
    "#         browser.execute_script(\"var t2 = document.getElementsByClassName('datepickerto')[0]; t2.value='\" +str(enstr)+ \"';\");\n",
    "\n",
    "    #how many people\n",
    "#     browser.execute_script(\"var t3 = document.getElementsByClassName('total_pass_input')[0]; t3.value=3;\");\n",
    "#     browser.execute_script(\"var t2 = document.getElementsByClassName('datepickerto')[0]; t2.value='2019/04/17';\");\n",
    "\n",
    "\n",
    "    browser.find_element_by_css_selector('.btn.btn-primary').submit()\n",
    "    time.sleep(10)\n",
    "\n",
    "    try:\n",
    "        if browser.find_element_by_class_name('error-title') != []:\n",
    "            print(browser.find_element_by_class_name('error-title').text)\n",
    "    except:\n",
    "        time.sleep(10)\n",
    "        print('no error')\n",
    "        html = browser.page_source\n",
    "        soup = bs(html)\n",
    "\n",
    "except Exception as e:\n",
    "    print(str(e))\n",
    "    pass\n",
    "\n",
    "browser.quit()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "菲律賓 - 所有機場 XXX\n",
    "宿霧 (CEB) CEB\n",
    "卡里博(長灘島) (KLO) KLO\n",
    "巴拉望(公主港) (PPS) PPS\n",
    "北海道地區 - 所有機場 XX2\n",
    "旭川 (AKJ) AKJ\n",
    "函館 (HKD) HKD\n",
    "東北地區 - 所有機場 XX5\n",
    "花卷 (HNA) HNA\n",
    "仙台 (SDJ) SDJ\n",
    "中部地區 - 所有機場 XX6\n",
    "小松 (KMQ) KMQ\n",
    "名古屋 (NGO) NGO\n",
    "關東地區 - 所有機場 XX7\n",
    "東京羽田 (HND) HND\n",
    "茨城 (IBR) IBR\n",
    "東京成田 (NRT) NRT\n",
    "關西地區 - 所有機場 XX8\n",
    "大阪 (KIX) KIX\n",
    "岡山 (OKJ) OKJ\n",
    "九州地區 - 所有機場 XX9\n",
    "福岡 (FUK) FUK\n",
    "佐賀 (HSG) HSG\n",
    "沖繩 (OKA) OKA\n",
    "南韓 - 所有機場 XX4\n",
    "濟州 (CJU) CJU\n",
    "釜山 (PUS) PUS\n",
    "大邱 (TAE) TAE\n",
    "台灣 - 所有機場 XX3\n",
    "台北(桃園) (TPE) TPE\n",
    "台中 (RMQ) RMQ\n",
    "高雄 (KHH) KHH\n",
    "曼谷(廊曼) (DMK) DMK\n",
    "澳門 (MFM) MFM"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "metadata": {},
   "outputs": [],
   "source": [
    "def scrapeJob(afterday=0, daylength=0,fromwhere='XX3',towhere='XXX'):\n",
    "    today = datetime.datetime.now() + datetime.timedelta(days=afterday)\n",
    "    tostr = today.strftime('%Y/%m/%d')\n",
    "    end_date = today + datetime.timedelta(days=daylength) + datetime.timedelta(days=afterday)\n",
    "    enstr = end_date.strftime('%Y/%m/%d')\n",
    "\n",
    "    browser = webdriver.Chrome(executable_path='driver/chromedriver')\n",
    "    browser.get('https://www.tigerairtw.com/zh-tw')\n",
    "#     print(tostr)\n",
    "    \n",
    "\n",
    "    try:\n",
    "        time.sleep(4)\n",
    "        #checkbox\n",
    "        if daylength == 0:\n",
    "            browser.execute_script(\"document.getElementsByClassName('class_type')[1].setAttribute('checked', true)\")\n",
    "    #         checkboxs=browser.find_elements_by_class_name('class_type')\n",
    "    #         for checkbox in checkboxs:\n",
    "    #             print(checkbox.get_attribute('value'),checkbox.get_attribute('checked')\n",
    "\n",
    "        # select part\n",
    "        s1 = Select(browser.find_element_by_class_name('selectOrigin'))\n",
    "    #     for select in s1.options:\n",
    "    #         print(select.text, select.get_attribute('value'))\n",
    "        s1.select_by_value(fromwhere)\n",
    "        s2 = Select(browser.find_element_by_class_name('selectDestination'))\n",
    "        # for select in s2.options:\n",
    "        #     print(select.text)\n",
    "        s2.select_by_value(towhere)\n",
    "\n",
    "        # The data table\n",
    "        browser.execute_script(\"var t1 = document.getElementsByClassName('datepickerfrom')[0]; t1.value='\" +str(tostr)+ \"';\");\n",
    "        if daylength != 0:\n",
    "            browser.execute_script(\"var t2 = document.getElementsByClassName('datepickerto')[0]; t2.value='\" +str(enstr)+ \"';\");\n",
    "\n",
    "        #how many people\n",
    "    #     browser.execute_script(\"var t3 = document.getElementsByClassName('total_pass_input')[0]; t3.value=3;\");\n",
    "    #     browser.execute_script(\"var t2 = document.getElementsByClassName('datepickerto')[0]; t2.value='2019/04/17';\");\n",
    "\n",
    "\n",
    "        browser.find_element_by_css_selector('.btn.btn-primary').submit()\n",
    "        time.sleep(10)\n",
    "\n",
    "        try:\n",
    "            if browser.find_element_by_class_name('error-title') != []:\n",
    "                print(tostr,enstr)\n",
    "                print(browser.find_element_by_class_name('error-title').text)\n",
    "        except:\n",
    "            print(tostr,enstr)\n",
    "            print('no error')\n",
    "            html = browser.page_source\n",
    "            soup = bs(html)\n",
    "\n",
    "    except Exception as e:\n",
    "        print(str(e))\n",
    "        pass\n",
    "    browser.quit()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 52,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2019/05/18 2019/06/23\n",
      "page not found\n",
      "2019/05/19 2019/06/25\n",
      "page not found\n",
      "2019/05/21 2019/06/29\n",
      "page not found\n",
      "2019/05/20 2019/06/27\n",
      "page not found\n",
      "2019/05/23 2019/07/03\n",
      "page not found\n",
      "2019/05/24 2019/07/05\n",
      "page not found\n",
      "2019/05/22 2019/07/01\n",
      "page not found\n",
      "2019/05/25 2019/07/07\n",
      "page not found\n",
      "2019/05/28 2019/07/13\n",
      "page not found\n",
      "2019/06/02 2019/07/23\n",
      "page not found\n",
      "2019/05/29 2019/07/15\n",
      "page not found\n",
      "2019/06/04 2019/07/27\n",
      "page not found\n",
      "2019/06/05 2019/07/29\n",
      "page not found\n",
      "2019/06/07 2019/08/02\n",
      "page not found\n",
      "2019/05/31 2019/07/19\n",
      "page not found\n",
      "2019/06/08 2019/08/04\n",
      "page not found\n",
      "2019/06/09 2019/08/06\n",
      "page not found\n",
      "2019/06/10 2019/08/08\n",
      "page not found\n",
      "2019/06/11 2019/08/10\n",
      "no error\n",
      "2019/06/14 2019/08/16\n",
      "page not found\n",
      "2019/06/15 2019/08/18\n",
      "page not found\n",
      "2019/06/16 2019/08/20\n",
      "no error\n",
      "2019/06/06 2019/07/31\n",
      "no error\n",
      "2019/06/17 2019/08/22\n",
      "no error\n",
      "2019/06/20 2019/08/28\n",
      "no error\n",
      "2019/06/18 2019/08/24\n",
      "no error\n",
      "2019/06/01 2019/07/21\n",
      "no error\n",
      "('Connection aborted.', RemoteDisconnected('Remote end closed connection without response',))\n",
      "('Connection aborted.', RemoteDisconnected('Remote end closed connection without response',))\n",
      "('Connection aborted.', RemoteDisconnected('Remote end closed connection without response',))\n",
      "('Connection aborted.', RemoteDisconnected('Remote end closed connection without response',))\n",
      "('Connection aborted.', RemoteDisconnected('Remote end closed connection without response',))\n",
      "('Connection aborted.', RemoteDisconnected('Remote end closed connection without response',))\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Exception in thread Thread-284:\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/james/.local/lib/python3.6/site-packages/urllib3/connectionpool.py\", line 600, in urlopen\n",
      "    chunked=chunked)\n",
      "  File \"/home/james/.local/lib/python3.6/site-packages/urllib3/connectionpool.py\", line 384, in _make_request\n",
      "    six.raise_from(e, None)\n",
      "  File \"<string>\", line 2, in raise_from\n",
      "  File \"/home/james/.local/lib/python3.6/site-packages/urllib3/connectionpool.py\", line 380, in _make_request\n",
      "    httplib_response = conn.getresponse()\n",
      "  File \"/usr/lib/python3.6/http/client.py\", line 1331, in getresponse\n",
      "    response.begin()\n",
      "  File \"/usr/lib/python3.6/http/client.py\", line 297, in begin\n",
      "    version, status, reason = self._read_status()\n",
      "  File \"/usr/lib/python3.6/http/client.py\", line 266, in _read_status\n",
      "    raise RemoteDisconnected(\"Remote end closed connection without\"\n",
      "http.client.RemoteDisconnected: Remote end closed connection without response\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"/usr/lib/python3.6/threading.py\", line 916, in _bootstrap_inner\n",
      "    self.run()\n",
      "  File \"/usr/lib/python3.6/threading.py\", line 864, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"<ipython-input-49-a61b01ac7f52>\", line 8, in scrapeJob\n",
      "    browser.get('https://www.tigerairtw.com/zh-tw')\n",
      "  File \"/home/james/.local/lib/python3.6/site-packages/selenium/webdriver/remote/webdriver.py\", line 333, in get\n",
      "    self.execute(Command.GET, {'url': url})\n",
      "  File \"/home/james/.local/lib/python3.6/site-packages/selenium/webdriver/remote/webdriver.py\", line 319, in execute\n",
      "    response = self.command_executor.execute(driver_command, params)\n",
      "  File \"/home/james/.local/lib/python3.6/site-packages/selenium/webdriver/remote/remote_connection.py\", line 374, in execute\n",
      "    return self._request(command_info[0], url, body=data)\n",
      "  File \"/home/james/.local/lib/python3.6/site-packages/selenium/webdriver/remote/remote_connection.py\", line 397, in _request\n",
      "    resp = self._conn.request(method, url, body=body, headers=headers)\n",
      "  File \"/home/james/.local/lib/python3.6/site-packages/urllib3/request.py\", line 72, in request\n",
      "    **urlopen_kw)\n",
      "  File \"/home/james/.local/lib/python3.6/site-packages/urllib3/request.py\", line 150, in request_encode_body\n",
      "    return self.urlopen(method, url, **extra_kw)\n",
      "  File \"/home/james/.local/lib/python3.6/site-packages/urllib3/poolmanager.py\", line 323, in urlopen\n",
      "    response = conn.urlopen(method, u.request_uri, **kw)\n",
      "  File \"/home/james/.local/lib/python3.6/site-packages/urllib3/connectionpool.py\", line 638, in urlopen\n",
      "    _stacktrace=sys.exc_info()[2])\n",
      "  File \"/home/james/.local/lib/python3.6/site-packages/urllib3/util/retry.py\", line 367, in increment\n",
      "    raise six.reraise(type(error), error, _stacktrace)\n",
      "  File \"/home/james/.local/lib/python3.6/site-packages/urllib3/packages/six.py\", line 685, in reraise\n",
      "    raise value.with_traceback(tb)\n",
      "  File \"/home/james/.local/lib/python3.6/site-packages/urllib3/connectionpool.py\", line 600, in urlopen\n",
      "    chunked=chunked)\n",
      "  File \"/home/james/.local/lib/python3.6/site-packages/urllib3/connectionpool.py\", line 384, in _make_request\n",
      "    six.raise_from(e, None)\n",
      "  File \"<string>\", line 2, in raise_from\n",
      "  File \"/home/james/.local/lib/python3.6/site-packages/urllib3/connectionpool.py\", line 380, in _make_request\n",
      "    httplib_response = conn.getresponse()\n",
      "  File \"/usr/lib/python3.6/http/client.py\", line 1331, in getresponse\n",
      "    response.begin()\n",
      "  File \"/usr/lib/python3.6/http/client.py\", line 297, in begin\n",
      "    version, status, reason = self._read_status()\n",
      "  File \"/usr/lib/python3.6/http/client.py\", line 266, in _read_status\n",
      "    raise RemoteDisconnected(\"Remote end closed connection without\"\n",
      "urllib3.exceptions.ProtocolError: ('Connection aborted.', RemoteDisconnected('Remote end closed connection without response',))\n",
      "\n",
      "Exception in thread Thread-316:\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/james/.local/lib/python3.6/site-packages/urllib3/connectionpool.py\", line 600, in urlopen\n",
      "    chunked=chunked)\n",
      "  File \"/home/james/.local/lib/python3.6/site-packages/urllib3/connectionpool.py\", line 384, in _make_request\n",
      "    six.raise_from(e, None)\n",
      "  File \"<string>\", line 2, in raise_from\n",
      "  File \"/home/james/.local/lib/python3.6/site-packages/urllib3/connectionpool.py\", line 380, in _make_request\n",
      "    httplib_response = conn.getresponse()\n",
      "  File \"/usr/lib/python3.6/http/client.py\", line 1331, in getresponse\n",
      "    response.begin()\n",
      "  File \"/usr/lib/python3.6/http/client.py\", line 297, in begin\n",
      "    version, status, reason = self._read_status()\n",
      "  File \"/usr/lib/python3.6/http/client.py\", line 266, in _read_status\n",
      "    raise RemoteDisconnected(\"Remote end closed connection without\"\n",
      "http.client.RemoteDisconnected: Remote end closed connection without response\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"/usr/lib/python3.6/threading.py\", line 916, in _bootstrap_inner\n",
      "    self.run()\n",
      "  File \"/usr/lib/python3.6/threading.py\", line 864, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"<ipython-input-49-a61b01ac7f52>\", line 8, in scrapeJob\n",
      "    browser.get('https://www.tigerairtw.com/zh-tw')\n",
      "  File \"/home/james/.local/lib/python3.6/site-packages/selenium/webdriver/remote/webdriver.py\", line 333, in get\n",
      "    self.execute(Command.GET, {'url': url})\n",
      "  File \"/home/james/.local/lib/python3.6/site-packages/selenium/webdriver/remote/webdriver.py\", line 319, in execute\n",
      "    response = self.command_executor.execute(driver_command, params)\n",
      "  File \"/home/james/.local/lib/python3.6/site-packages/selenium/webdriver/remote/remote_connection.py\", line 374, in execute\n",
      "    return self._request(command_info[0], url, body=data)\n",
      "  File \"/home/james/.local/lib/python3.6/site-packages/selenium/webdriver/remote/remote_connection.py\", line 397, in _request\n",
      "    resp = self._conn.request(method, url, body=body, headers=headers)\n",
      "  File \"/home/james/.local/lib/python3.6/site-packages/urllib3/request.py\", line 72, in request\n",
      "    **urlopen_kw)\n",
      "  File \"/home/james/.local/lib/python3.6/site-packages/urllib3/request.py\", line 150, in request_encode_body\n",
      "    return self.urlopen(method, url, **extra_kw)\n",
      "  File \"/home/james/.local/lib/python3.6/site-packages/urllib3/poolmanager.py\", line 323, in urlopen\n",
      "    response = conn.urlopen(method, u.request_uri, **kw)\n",
      "  File \"/home/james/.local/lib/python3.6/site-packages/urllib3/connectionpool.py\", line 638, in urlopen\n",
      "    _stacktrace=sys.exc_info()[2])\n",
      "  File \"/home/james/.local/lib/python3.6/site-packages/urllib3/util/retry.py\", line 367, in increment\n",
      "    raise six.reraise(type(error), error, _stacktrace)\n",
      "  File \"/home/james/.local/lib/python3.6/site-packages/urllib3/packages/six.py\", line 685, in reraise\n",
      "    raise value.with_traceback(tb)\n",
      "  File \"/home/james/.local/lib/python3.6/site-packages/urllib3/connectionpool.py\", line 600, in urlopen\n",
      "    chunked=chunked)\n",
      "  File \"/home/james/.local/lib/python3.6/site-packages/urllib3/connectionpool.py\", line 384, in _make_request\n",
      "    six.raise_from(e, None)\n",
      "  File \"<string>\", line 2, in raise_from\n",
      "  File \"/home/james/.local/lib/python3.6/site-packages/urllib3/connectionpool.py\", line 380, in _make_request\n",
      "    httplib_response = conn.getresponse()\n",
      "  File \"/usr/lib/python3.6/http/client.py\", line 1331, in getresponse\n",
      "    response.begin()\n",
      "  File \"/usr/lib/python3.6/http/client.py\", line 297, in begin\n",
      "    version, status, reason = self._read_status()\n",
      "  File \"/usr/lib/python3.6/http/client.py\", line 266, in _read_status\n",
      "    raise RemoteDisconnected(\"Remote end closed connection without\"\n",
      "urllib3.exceptions.ProtocolError: ('Connection aborted.', RemoteDisconnected('Remote end closed connection without response',))\n",
      "\n",
      "Exception in thread Thread-309:\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/james/.local/lib/python3.6/site-packages/urllib3/connectionpool.py\", line 600, in urlopen\n",
      "    chunked=chunked)\n",
      "  File \"/home/james/.local/lib/python3.6/site-packages/urllib3/connectionpool.py\", line 384, in _make_request\n",
      "    six.raise_from(e, None)\n",
      "  File \"<string>\", line 2, in raise_from\n",
      "  File \"/home/james/.local/lib/python3.6/site-packages/urllib3/connectionpool.py\", line 380, in _make_request\n",
      "    httplib_response = conn.getresponse()\n",
      "  File \"/usr/lib/python3.6/http/client.py\", line 1331, in getresponse\n",
      "    response.begin()\n",
      "  File \"/usr/lib/python3.6/http/client.py\", line 297, in begin\n",
      "    version, status, reason = self._read_status()\n",
      "  File \"/usr/lib/python3.6/http/client.py\", line 266, in _read_status\n",
      "    raise RemoteDisconnected(\"Remote end closed connection without\"\n",
      "http.client.RemoteDisconnected: Remote end closed connection without response\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"/usr/lib/python3.6/threading.py\", line 916, in _bootstrap_inner\n",
      "    self.run()\n",
      "  File \"/usr/lib/python3.6/threading.py\", line 864, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"<ipython-input-49-a61b01ac7f52>\", line 8, in scrapeJob\n",
      "    browser.get('https://www.tigerairtw.com/zh-tw')\n",
      "  File \"/home/james/.local/lib/python3.6/site-packages/selenium/webdriver/remote/webdriver.py\", line 333, in get\n",
      "    self.execute(Command.GET, {'url': url})\n",
      "  File \"/home/james/.local/lib/python3.6/site-packages/selenium/webdriver/remote/webdriver.py\", line 319, in execute\n",
      "    response = self.command_executor.execute(driver_command, params)\n",
      "  File \"/home/james/.local/lib/python3.6/site-packages/selenium/webdriver/remote/remote_connection.py\", line 374, in execute\n",
      "    return self._request(command_info[0], url, body=data)\n",
      "  File \"/home/james/.local/lib/python3.6/site-packages/selenium/webdriver/remote/remote_connection.py\", line 397, in _request\n",
      "    resp = self._conn.request(method, url, body=body, headers=headers)\n",
      "  File \"/home/james/.local/lib/python3.6/site-packages/urllib3/request.py\", line 72, in request\n",
      "    **urlopen_kw)\n",
      "  File \"/home/james/.local/lib/python3.6/site-packages/urllib3/request.py\", line 150, in request_encode_body\n",
      "    return self.urlopen(method, url, **extra_kw)\n",
      "  File \"/home/james/.local/lib/python3.6/site-packages/urllib3/poolmanager.py\", line 323, in urlopen\n",
      "    response = conn.urlopen(method, u.request_uri, **kw)\n",
      "  File \"/home/james/.local/lib/python3.6/site-packages/urllib3/connectionpool.py\", line 638, in urlopen\n",
      "    _stacktrace=sys.exc_info()[2])\n",
      "  File \"/home/james/.local/lib/python3.6/site-packages/urllib3/util/retry.py\", line 367, in increment\n",
      "    raise six.reraise(type(error), error, _stacktrace)\n",
      "  File \"/home/james/.local/lib/python3.6/site-packages/urllib3/packages/six.py\", line 685, in reraise\n",
      "    raise value.with_traceback(tb)\n",
      "  File \"/home/james/.local/lib/python3.6/site-packages/urllib3/connectionpool.py\", line 600, in urlopen\n",
      "    chunked=chunked)\n",
      "  File \"/home/james/.local/lib/python3.6/site-packages/urllib3/connectionpool.py\", line 384, in _make_request\n",
      "    six.raise_from(e, None)\n",
      "  File \"<string>\", line 2, in raise_from\n",
      "  File \"/home/james/.local/lib/python3.6/site-packages/urllib3/connectionpool.py\", line 380, in _make_request\n",
      "    httplib_response = conn.getresponse()\n",
      "  File \"/usr/lib/python3.6/http/client.py\", line 1331, in getresponse\n",
      "    response.begin()\n",
      "  File \"/usr/lib/python3.6/http/client.py\", line 297, in begin\n",
      "    version, status, reason = self._read_status()\n",
      "  File \"/usr/lib/python3.6/http/client.py\", line 266, in _read_status\n",
      "    raise RemoteDisconnected(\"Remote end closed connection without\"\n",
      "urllib3.exceptions.ProtocolError: ('Connection aborted.', RemoteDisconnected('Remote end closed connection without response',))\n",
      "\n",
      "Exception in thread Thread-280:\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/james/.local/lib/python3.6/site-packages/urllib3/connectionpool.py\", line 600, in urlopen\n",
      "    chunked=chunked)\n",
      "  File \"/home/james/.local/lib/python3.6/site-packages/urllib3/connectionpool.py\", line 384, in _make_request\n",
      "    six.raise_from(e, None)\n",
      "  File \"<string>\", line 2, in raise_from\n",
      "  File \"/home/james/.local/lib/python3.6/site-packages/urllib3/connectionpool.py\", line 380, in _make_request\n",
      "    httplib_response = conn.getresponse()\n",
      "  File \"/usr/lib/python3.6/http/client.py\", line 1331, in getresponse\n",
      "    response.begin()\n",
      "  File \"/usr/lib/python3.6/http/client.py\", line 297, in begin\n",
      "    version, status, reason = self._read_status()\n",
      "  File \"/usr/lib/python3.6/http/client.py\", line 266, in _read_status\n",
      "    raise RemoteDisconnected(\"Remote end closed connection without\"\n",
      "http.client.RemoteDisconnected: Remote end closed connection without response\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"/usr/lib/python3.6/threading.py\", line 916, in _bootstrap_inner\n",
      "    self.run()\n",
      "  File \"/usr/lib/python3.6/threading.py\", line 864, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"<ipython-input-49-a61b01ac7f52>\", line 8, in scrapeJob\n",
      "    browser.get('https://www.tigerairtw.com/zh-tw')\n",
      "  File \"/home/james/.local/lib/python3.6/site-packages/selenium/webdriver/remote/webdriver.py\", line 333, in get\n",
      "    self.execute(Command.GET, {'url': url})\n",
      "  File \"/home/james/.local/lib/python3.6/site-packages/selenium/webdriver/remote/webdriver.py\", line 319, in execute\n",
      "    response = self.command_executor.execute(driver_command, params)\n",
      "  File \"/home/james/.local/lib/python3.6/site-packages/selenium/webdriver/remote/remote_connection.py\", line 374, in execute\n",
      "    return self._request(command_info[0], url, body=data)\n",
      "  File \"/home/james/.local/lib/python3.6/site-packages/selenium/webdriver/remote/remote_connection.py\", line 397, in _request\n",
      "    resp = self._conn.request(method, url, body=body, headers=headers)\n",
      "  File \"/home/james/.local/lib/python3.6/site-packages/urllib3/request.py\", line 72, in request\n",
      "    **urlopen_kw)\n",
      "  File \"/home/james/.local/lib/python3.6/site-packages/urllib3/request.py\", line 150, in request_encode_body\n",
      "    return self.urlopen(method, url, **extra_kw)\n",
      "  File \"/home/james/.local/lib/python3.6/site-packages/urllib3/poolmanager.py\", line 323, in urlopen\n",
      "    response = conn.urlopen(method, u.request_uri, **kw)\n",
      "  File \"/home/james/.local/lib/python3.6/site-packages/urllib3/connectionpool.py\", line 638, in urlopen\n",
      "    _stacktrace=sys.exc_info()[2])\n",
      "  File \"/home/james/.local/lib/python3.6/site-packages/urllib3/util/retry.py\", line 367, in increment\n",
      "    raise six.reraise(type(error), error, _stacktrace)\n",
      "  File \"/home/james/.local/lib/python3.6/site-packages/urllib3/packages/six.py\", line 685, in reraise\n",
      "    raise value.with_traceback(tb)\n",
      "  File \"/home/james/.local/lib/python3.6/site-packages/urllib3/connectionpool.py\", line 600, in urlopen\n",
      "    chunked=chunked)\n",
      "  File \"/home/james/.local/lib/python3.6/site-packages/urllib3/connectionpool.py\", line 384, in _make_request\n",
      "    six.raise_from(e, None)\n",
      "  File \"<string>\", line 2, in raise_from\n",
      "  File \"/home/james/.local/lib/python3.6/site-packages/urllib3/connectionpool.py\", line 380, in _make_request\n",
      "    httplib_response = conn.getresponse()\n",
      "  File \"/usr/lib/python3.6/http/client.py\", line 1331, in getresponse\n",
      "    response.begin()\n",
      "  File \"/usr/lib/python3.6/http/client.py\", line 297, in begin\n",
      "    version, status, reason = self._read_status()\n",
      "  File \"/usr/lib/python3.6/http/client.py\", line 266, in _read_status\n",
      "    raise RemoteDisconnected(\"Remote end closed connection without\"\n",
      "urllib3.exceptions.ProtocolError: ('Connection aborted.', RemoteDisconnected('Remote end closed connection without response',))\n",
      "\n",
      "Exception in thread Thread-276:\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/james/.local/lib/python3.6/site-packages/urllib3/connectionpool.py\", line 600, in urlopen\n",
      "    chunked=chunked)\n",
      "  File \"/home/james/.local/lib/python3.6/site-packages/urllib3/connectionpool.py\", line 384, in _make_request\n",
      "    six.raise_from(e, None)\n",
      "  File \"<string>\", line 2, in raise_from\n",
      "  File \"/home/james/.local/lib/python3.6/site-packages/urllib3/connectionpool.py\", line 380, in _make_request\n",
      "    httplib_response = conn.getresponse()\n",
      "  File \"/usr/lib/python3.6/http/client.py\", line 1331, in getresponse\n",
      "    response.begin()\n",
      "  File \"/usr/lib/python3.6/http/client.py\", line 297, in begin\n",
      "    version, status, reason = self._read_status()\n",
      "  File \"/usr/lib/python3.6/http/client.py\", line 266, in _read_status\n",
      "    raise RemoteDisconnected(\"Remote end closed connection without\"\n",
      "http.client.RemoteDisconnected: Remote end closed connection without response\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"/usr/lib/python3.6/threading.py\", line 916, in _bootstrap_inner\n",
      "    self.run()\n",
      "  File \"/usr/lib/python3.6/threading.py\", line 864, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"<ipython-input-49-a61b01ac7f52>\", line 8, in scrapeJob\n",
      "    browser.get('https://www.tigerairtw.com/zh-tw')\n",
      "  File \"/home/james/.local/lib/python3.6/site-packages/selenium/webdriver/remote/webdriver.py\", line 333, in get\n",
      "    self.execute(Command.GET, {'url': url})\n",
      "  File \"/home/james/.local/lib/python3.6/site-packages/selenium/webdriver/remote/webdriver.py\", line 319, in execute\n",
      "    response = self.command_executor.execute(driver_command, params)\n",
      "  File \"/home/james/.local/lib/python3.6/site-packages/selenium/webdriver/remote/remote_connection.py\", line 374, in execute\n",
      "    return self._request(command_info[0], url, body=data)\n",
      "  File \"/home/james/.local/lib/python3.6/site-packages/selenium/webdriver/remote/remote_connection.py\", line 397, in _request\n",
      "    resp = self._conn.request(method, url, body=body, headers=headers)\n",
      "  File \"/home/james/.local/lib/python3.6/site-packages/urllib3/request.py\", line 72, in request\n",
      "    **urlopen_kw)\n",
      "  File \"/home/james/.local/lib/python3.6/site-packages/urllib3/request.py\", line 150, in request_encode_body\n",
      "    return self.urlopen(method, url, **extra_kw)\n",
      "  File \"/home/james/.local/lib/python3.6/site-packages/urllib3/poolmanager.py\", line 323, in urlopen\n",
      "    response = conn.urlopen(method, u.request_uri, **kw)\n",
      "  File \"/home/james/.local/lib/python3.6/site-packages/urllib3/connectionpool.py\", line 638, in urlopen\n",
      "    _stacktrace=sys.exc_info()[2])\n",
      "  File \"/home/james/.local/lib/python3.6/site-packages/urllib3/util/retry.py\", line 367, in increment\n",
      "    raise six.reraise(type(error), error, _stacktrace)\n",
      "  File \"/home/james/.local/lib/python3.6/site-packages/urllib3/packages/six.py\", line 685, in reraise\n",
      "    raise value.with_traceback(tb)\n",
      "  File \"/home/james/.local/lib/python3.6/site-packages/urllib3/connectionpool.py\", line 600, in urlopen\n",
      "    chunked=chunked)\n",
      "  File \"/home/james/.local/lib/python3.6/site-packages/urllib3/connectionpool.py\", line 384, in _make_request\n",
      "    six.raise_from(e, None)\n",
      "  File \"<string>\", line 2, in raise_from\n",
      "  File \"/home/james/.local/lib/python3.6/site-packages/urllib3/connectionpool.py\", line 380, in _make_request\n",
      "    httplib_response = conn.getresponse()\n",
      "  File \"/usr/lib/python3.6/http/client.py\", line 1331, in getresponse\n",
      "    response.begin()\n",
      "  File \"/usr/lib/python3.6/http/client.py\", line 297, in begin\n",
      "    version, status, reason = self._read_status()\n",
      "  File \"/usr/lib/python3.6/http/client.py\", line 266, in _read_status\n",
      "    raise RemoteDisconnected(\"Remote end closed connection without\"\n",
      "urllib3.exceptions.ProtocolError: ('Connection aborted.', RemoteDisconnected('Remote end closed connection without response',))\n",
      "\n",
      "Exception in thread Thread-293:\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/james/.local/lib/python3.6/site-packages/urllib3/connectionpool.py\", line 600, in urlopen\n",
      "    chunked=chunked)\n",
      "  File \"/home/james/.local/lib/python3.6/site-packages/urllib3/connectionpool.py\", line 384, in _make_request\n",
      "    six.raise_from(e, None)\n",
      "  File \"<string>\", line 2, in raise_from\n",
      "  File \"/home/james/.local/lib/python3.6/site-packages/urllib3/connectionpool.py\", line 380, in _make_request\n",
      "    httplib_response = conn.getresponse()\n",
      "  File \"/usr/lib/python3.6/http/client.py\", line 1331, in getresponse\n",
      "    response.begin()\n",
      "  File \"/usr/lib/python3.6/http/client.py\", line 297, in begin\n",
      "    version, status, reason = self._read_status()\n",
      "  File \"/usr/lib/python3.6/http/client.py\", line 266, in _read_status\n",
      "    raise RemoteDisconnected(\"Remote end closed connection without\"\n",
      "http.client.RemoteDisconnected: Remote end closed connection without response\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"/usr/lib/python3.6/threading.py\", line 916, in _bootstrap_inner\n",
      "    self.run()\n",
      "  File \"/usr/lib/python3.6/threading.py\", line 864, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"<ipython-input-49-a61b01ac7f52>\", line 8, in scrapeJob\n",
      "    browser.get('https://www.tigerairtw.com/zh-tw')\n",
      "  File \"/home/james/.local/lib/python3.6/site-packages/selenium/webdriver/remote/webdriver.py\", line 333, in get\n",
      "    self.execute(Command.GET, {'url': url})\n",
      "  File \"/home/james/.local/lib/python3.6/site-packages/selenium/webdriver/remote/webdriver.py\", line 319, in execute\n",
      "    response = self.command_executor.execute(driver_command, params)\n",
      "  File \"/home/james/.local/lib/python3.6/site-packages/selenium/webdriver/remote/remote_connection.py\", line 374, in execute\n",
      "    return self._request(command_info[0], url, body=data)\n",
      "  File \"/home/james/.local/lib/python3.6/site-packages/selenium/webdriver/remote/remote_connection.py\", line 397, in _request\n",
      "    resp = self._conn.request(method, url, body=body, headers=headers)\n",
      "  File \"/home/james/.local/lib/python3.6/site-packages/urllib3/request.py\", line 72, in request\n",
      "    **urlopen_kw)\n",
      "  File \"/home/james/.local/lib/python3.6/site-packages/urllib3/request.py\", line 150, in request_encode_body\n",
      "    return self.urlopen(method, url, **extra_kw)\n",
      "  File \"/home/james/.local/lib/python3.6/site-packages/urllib3/poolmanager.py\", line 323, in urlopen\n",
      "    response = conn.urlopen(method, u.request_uri, **kw)\n",
      "  File \"/home/james/.local/lib/python3.6/site-packages/urllib3/connectionpool.py\", line 638, in urlopen\n",
      "    _stacktrace=sys.exc_info()[2])\n",
      "  File \"/home/james/.local/lib/python3.6/site-packages/urllib3/util/retry.py\", line 367, in increment\n",
      "    raise six.reraise(type(error), error, _stacktrace)\n",
      "  File \"/home/james/.local/lib/python3.6/site-packages/urllib3/packages/six.py\", line 685, in reraise\n",
      "    raise value.with_traceback(tb)\n",
      "  File \"/home/james/.local/lib/python3.6/site-packages/urllib3/connectionpool.py\", line 600, in urlopen\n",
      "    chunked=chunked)\n",
      "  File \"/home/james/.local/lib/python3.6/site-packages/urllib3/connectionpool.py\", line 384, in _make_request\n",
      "    six.raise_from(e, None)\n",
      "  File \"<string>\", line 2, in raise_from\n",
      "  File \"/home/james/.local/lib/python3.6/site-packages/urllib3/connectionpool.py\", line 380, in _make_request\n",
      "    httplib_response = conn.getresponse()\n",
      "  File \"/usr/lib/python3.6/http/client.py\", line 1331, in getresponse\n",
      "    response.begin()\n",
      "  File \"/usr/lib/python3.6/http/client.py\", line 297, in begin\n",
      "    version, status, reason = self._read_status()\n",
      "  File \"/usr/lib/python3.6/http/client.py\", line 266, in _read_status\n",
      "    raise RemoteDisconnected(\"Remote end closed connection without\"\n",
      "urllib3.exceptions.ProtocolError: ('Connection aborted.', RemoteDisconnected('Remote end closed connection without response',))\n",
      "\n",
      "Exception in thread Thread-311:\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/james/.local/lib/python3.6/site-packages/urllib3/connectionpool.py\", line 600, in urlopen\n",
      "    chunked=chunked)\n",
      "  File \"/home/james/.local/lib/python3.6/site-packages/urllib3/connectionpool.py\", line 384, in _make_request\n",
      "    six.raise_from(e, None)\n",
      "  File \"<string>\", line 2, in raise_from\n",
      "  File \"/home/james/.local/lib/python3.6/site-packages/urllib3/connectionpool.py\", line 380, in _make_request\n",
      "    httplib_response = conn.getresponse()\n",
      "  File \"/usr/lib/python3.6/http/client.py\", line 1331, in getresponse\n",
      "    response.begin()\n",
      "  File \"/usr/lib/python3.6/http/client.py\", line 297, in begin\n",
      "    version, status, reason = self._read_status()\n",
      "  File \"/usr/lib/python3.6/http/client.py\", line 266, in _read_status\n",
      "    raise RemoteDisconnected(\"Remote end closed connection without\"\n",
      "http.client.RemoteDisconnected: Remote end closed connection without response\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"/usr/lib/python3.6/threading.py\", line 916, in _bootstrap_inner\n",
      "    self.run()\n",
      "  File \"/usr/lib/python3.6/threading.py\", line 864, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"<ipython-input-49-a61b01ac7f52>\", line 8, in scrapeJob\n",
      "    browser.get('https://www.tigerairtw.com/zh-tw')\n",
      "  File \"/home/james/.local/lib/python3.6/site-packages/selenium/webdriver/remote/webdriver.py\", line 333, in get\n",
      "    self.execute(Command.GET, {'url': url})\n",
      "  File \"/home/james/.local/lib/python3.6/site-packages/selenium/webdriver/remote/webdriver.py\", line 319, in execute\n",
      "    response = self.command_executor.execute(driver_command, params)\n",
      "  File \"/home/james/.local/lib/python3.6/site-packages/selenium/webdriver/remote/remote_connection.py\", line 374, in execute\n",
      "    return self._request(command_info[0], url, body=data)\n",
      "  File \"/home/james/.local/lib/python3.6/site-packages/selenium/webdriver/remote/remote_connection.py\", line 397, in _request\n",
      "    resp = self._conn.request(method, url, body=body, headers=headers)\n",
      "  File \"/home/james/.local/lib/python3.6/site-packages/urllib3/request.py\", line 72, in request\n",
      "    **urlopen_kw)\n",
      "  File \"/home/james/.local/lib/python3.6/site-packages/urllib3/request.py\", line 150, in request_encode_body\n",
      "    return self.urlopen(method, url, **extra_kw)\n",
      "  File \"/home/james/.local/lib/python3.6/site-packages/urllib3/poolmanager.py\", line 323, in urlopen\n",
      "    response = conn.urlopen(method, u.request_uri, **kw)\n",
      "  File \"/home/james/.local/lib/python3.6/site-packages/urllib3/connectionpool.py\", line 638, in urlopen\n",
      "    _stacktrace=sys.exc_info()[2])\n",
      "  File \"/home/james/.local/lib/python3.6/site-packages/urllib3/util/retry.py\", line 367, in increment\n",
      "    raise six.reraise(type(error), error, _stacktrace)\n",
      "  File \"/home/james/.local/lib/python3.6/site-packages/urllib3/packages/six.py\", line 685, in reraise\n",
      "    raise value.with_traceback(tb)\n",
      "  File \"/home/james/.local/lib/python3.6/site-packages/urllib3/connectionpool.py\", line 600, in urlopen\n",
      "    chunked=chunked)\n",
      "  File \"/home/james/.local/lib/python3.6/site-packages/urllib3/connectionpool.py\", line 384, in _make_request\n",
      "    six.raise_from(e, None)\n",
      "  File \"<string>\", line 2, in raise_from\n",
      "  File \"/home/james/.local/lib/python3.6/site-packages/urllib3/connectionpool.py\", line 380, in _make_request\n",
      "    httplib_response = conn.getresponse()\n",
      "  File \"/usr/lib/python3.6/http/client.py\", line 1331, in getresponse\n",
      "    response.begin()\n",
      "  File \"/usr/lib/python3.6/http/client.py\", line 297, in begin\n",
      "    version, status, reason = self._read_status()\n",
      "  File \"/usr/lib/python3.6/http/client.py\", line 266, in _read_status\n",
      "    raise RemoteDisconnected(\"Remote end closed connection without\"\n",
      "urllib3.exceptions.ProtocolError: ('Connection aborted.', RemoteDisconnected('Remote end closed connection without response',))\n",
      "\n",
      "Exception in thread Thread-318:\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/james/.local/lib/python3.6/site-packages/urllib3/connectionpool.py\", line 600, in urlopen\n",
      "    chunked=chunked)\n",
      "  File \"/home/james/.local/lib/python3.6/site-packages/urllib3/connectionpool.py\", line 384, in _make_request\n",
      "    six.raise_from(e, None)\n",
      "  File \"<string>\", line 2, in raise_from\n",
      "  File \"/home/james/.local/lib/python3.6/site-packages/urllib3/connectionpool.py\", line 380, in _make_request\n",
      "    httplib_response = conn.getresponse()\n",
      "  File \"/usr/lib/python3.6/http/client.py\", line 1331, in getresponse\n",
      "    response.begin()\n",
      "  File \"/usr/lib/python3.6/http/client.py\", line 297, in begin\n",
      "    version, status, reason = self._read_status()\n",
      "  File \"/usr/lib/python3.6/http/client.py\", line 266, in _read_status\n",
      "    raise RemoteDisconnected(\"Remote end closed connection without\"\n",
      "http.client.RemoteDisconnected: Remote end closed connection without response\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"/usr/lib/python3.6/threading.py\", line 916, in _bootstrap_inner\n",
      "    self.run()\n",
      "  File \"/usr/lib/python3.6/threading.py\", line 864, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"<ipython-input-49-a61b01ac7f52>\", line 7, in scrapeJob\n",
      "    browser = webdriver.Chrome(executable_path='driver/chromedriver')\n",
      "  File \"/home/james/.local/lib/python3.6/site-packages/selenium/webdriver/chrome/webdriver.py\", line 81, in __init__\n",
      "    desired_capabilities=desired_capabilities)\n",
      "  File \"/home/james/.local/lib/python3.6/site-packages/selenium/webdriver/remote/webdriver.py\", line 157, in __init__\n",
      "    self.start_session(capabilities, browser_profile)\n",
      "  File \"/home/james/.local/lib/python3.6/site-packages/selenium/webdriver/remote/webdriver.py\", line 252, in start_session\n",
      "    response = self.execute(Command.NEW_SESSION, parameters)\n",
      "  File \"/home/james/.local/lib/python3.6/site-packages/selenium/webdriver/remote/webdriver.py\", line 319, in execute\n",
      "    response = self.command_executor.execute(driver_command, params)\n",
      "  File \"/home/james/.local/lib/python3.6/site-packages/selenium/webdriver/remote/remote_connection.py\", line 374, in execute\n",
      "    return self._request(command_info[0], url, body=data)\n",
      "  File \"/home/james/.local/lib/python3.6/site-packages/selenium/webdriver/remote/remote_connection.py\", line 397, in _request\n",
      "    resp = self._conn.request(method, url, body=body, headers=headers)\n",
      "  File \"/home/james/.local/lib/python3.6/site-packages/urllib3/request.py\", line 72, in request\n",
      "    **urlopen_kw)\n",
      "  File \"/home/james/.local/lib/python3.6/site-packages/urllib3/request.py\", line 150, in request_encode_body\n",
      "    return self.urlopen(method, url, **extra_kw)\n",
      "  File \"/home/james/.local/lib/python3.6/site-packages/urllib3/poolmanager.py\", line 323, in urlopen\n",
      "    response = conn.urlopen(method, u.request_uri, **kw)\n",
      "  File \"/home/james/.local/lib/python3.6/site-packages/urllib3/connectionpool.py\", line 638, in urlopen\n",
      "    _stacktrace=sys.exc_info()[2])\n",
      "  File \"/home/james/.local/lib/python3.6/site-packages/urllib3/util/retry.py\", line 367, in increment\n",
      "    raise six.reraise(type(error), error, _stacktrace)\n",
      "  File \"/home/james/.local/lib/python3.6/site-packages/urllib3/packages/six.py\", line 685, in reraise\n",
      "    raise value.with_traceback(tb)\n",
      "  File \"/home/james/.local/lib/python3.6/site-packages/urllib3/connectionpool.py\", line 600, in urlopen\n",
      "    chunked=chunked)\n",
      "  File \"/home/james/.local/lib/python3.6/site-packages/urllib3/connectionpool.py\", line 384, in _make_request\n",
      "    six.raise_from(e, None)\n",
      "  File \"<string>\", line 2, in raise_from\n",
      "  File \"/home/james/.local/lib/python3.6/site-packages/urllib3/connectionpool.py\", line 380, in _make_request\n",
      "    httplib_response = conn.getresponse()\n",
      "  File \"/usr/lib/python3.6/http/client.py\", line 1331, in getresponse\n",
      "    response.begin()\n",
      "  File \"/usr/lib/python3.6/http/client.py\", line 297, in begin\n",
      "    version, status, reason = self._read_status()\n",
      "  File \"/usr/lib/python3.6/http/client.py\", line 266, in _read_status\n",
      "    raise RemoteDisconnected(\"Remote end closed connection without\"\n",
      "urllib3.exceptions.ProtocolError: ('Connection aborted.', RemoteDisconnected('Remote end closed connection without response',))\n",
      "\n",
      "Exception in thread Thread-305:\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/james/.local/lib/python3.6/site-packages/urllib3/connectionpool.py\", line 600, in urlopen\n",
      "    chunked=chunked)\n",
      "  File \"/home/james/.local/lib/python3.6/site-packages/urllib3/connectionpool.py\", line 384, in _make_request\n",
      "    six.raise_from(e, None)\n",
      "  File \"<string>\", line 2, in raise_from\n",
      "  File \"/home/james/.local/lib/python3.6/site-packages/urllib3/connectionpool.py\", line 380, in _make_request\n",
      "    httplib_response = conn.getresponse()\n",
      "  File \"/usr/lib/python3.6/http/client.py\", line 1331, in getresponse\n",
      "    response.begin()\n",
      "  File \"/usr/lib/python3.6/http/client.py\", line 297, in begin\n",
      "    version, status, reason = self._read_status()\n",
      "  File \"/usr/lib/python3.6/http/client.py\", line 266, in _read_status\n",
      "    raise RemoteDisconnected(\"Remote end closed connection without\"\n",
      "http.client.RemoteDisconnected: Remote end closed connection without response\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"/usr/lib/python3.6/threading.py\", line 916, in _bootstrap_inner\n",
      "    self.run()\n",
      "  File \"/usr/lib/python3.6/threading.py\", line 864, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"<ipython-input-49-a61b01ac7f52>\", line 8, in scrapeJob\n",
      "    browser.get('https://www.tigerairtw.com/zh-tw')\n",
      "  File \"/home/james/.local/lib/python3.6/site-packages/selenium/webdriver/remote/webdriver.py\", line 333, in get\n",
      "    self.execute(Command.GET, {'url': url})\n",
      "  File \"/home/james/.local/lib/python3.6/site-packages/selenium/webdriver/remote/webdriver.py\", line 319, in execute\n",
      "    response = self.command_executor.execute(driver_command, params)\n",
      "  File \"/home/james/.local/lib/python3.6/site-packages/selenium/webdriver/remote/remote_connection.py\", line 374, in execute\n",
      "    return self._request(command_info[0], url, body=data)\n",
      "  File \"/home/james/.local/lib/python3.6/site-packages/selenium/webdriver/remote/remote_connection.py\", line 397, in _request\n",
      "    resp = self._conn.request(method, url, body=body, headers=headers)\n",
      "  File \"/home/james/.local/lib/python3.6/site-packages/urllib3/request.py\", line 72, in request\n",
      "    **urlopen_kw)\n",
      "  File \"/home/james/.local/lib/python3.6/site-packages/urllib3/request.py\", line 150, in request_encode_body\n",
      "    return self.urlopen(method, url, **extra_kw)\n",
      "  File \"/home/james/.local/lib/python3.6/site-packages/urllib3/poolmanager.py\", line 323, in urlopen\n",
      "    response = conn.urlopen(method, u.request_uri, **kw)\n",
      "  File \"/home/james/.local/lib/python3.6/site-packages/urllib3/connectionpool.py\", line 638, in urlopen\n",
      "    _stacktrace=sys.exc_info()[2])\n",
      "  File \"/home/james/.local/lib/python3.6/site-packages/urllib3/util/retry.py\", line 367, in increment\n",
      "    raise six.reraise(type(error), error, _stacktrace)\n",
      "  File \"/home/james/.local/lib/python3.6/site-packages/urllib3/packages/six.py\", line 685, in reraise\n",
      "    raise value.with_traceback(tb)\n",
      "  File \"/home/james/.local/lib/python3.6/site-packages/urllib3/connectionpool.py\", line 600, in urlopen\n",
      "    chunked=chunked)\n",
      "  File \"/home/james/.local/lib/python3.6/site-packages/urllib3/connectionpool.py\", line 384, in _make_request\n",
      "    six.raise_from(e, None)\n",
      "  File \"<string>\", line 2, in raise_from\n",
      "  File \"/home/james/.local/lib/python3.6/site-packages/urllib3/connectionpool.py\", line 380, in _make_request\n",
      "    httplib_response = conn.getresponse()\n",
      "  File \"/usr/lib/python3.6/http/client.py\", line 1331, in getresponse\n",
      "    response.begin()\n",
      "  File \"/usr/lib/python3.6/http/client.py\", line 297, in begin\n",
      "    version, status, reason = self._read_status()\n",
      "  File \"/usr/lib/python3.6/http/client.py\", line 266, in _read_status\n",
      "    raise RemoteDisconnected(\"Remote end closed connection without\"\n",
      "urllib3.exceptions.ProtocolError: ('Connection aborted.', RemoteDisconnected('Remote end closed connection without response',))\n",
      "\n",
      "Exception in thread Thread-277:\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/james/.local/lib/python3.6/site-packages/urllib3/connectionpool.py\", line 600, in urlopen\n",
      "    chunked=chunked)\n",
      "  File \"/home/james/.local/lib/python3.6/site-packages/urllib3/connectionpool.py\", line 384, in _make_request\n",
      "    six.raise_from(e, None)\n",
      "  File \"<string>\", line 2, in raise_from\n",
      "  File \"/home/james/.local/lib/python3.6/site-packages/urllib3/connectionpool.py\", line 380, in _make_request\n",
      "    httplib_response = conn.getresponse()\n",
      "  File \"/usr/lib/python3.6/http/client.py\", line 1331, in getresponse\n",
      "    response.begin()\n",
      "  File \"/usr/lib/python3.6/http/client.py\", line 297, in begin\n",
      "    version, status, reason = self._read_status()\n",
      "  File \"/usr/lib/python3.6/http/client.py\", line 266, in _read_status\n",
      "    raise RemoteDisconnected(\"Remote end closed connection without\"\n",
      "http.client.RemoteDisconnected: Remote end closed connection without response\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"/usr/lib/python3.6/threading.py\", line 916, in _bootstrap_inner\n",
      "    self.run()\n",
      "  File \"/usr/lib/python3.6/threading.py\", line 864, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"<ipython-input-49-a61b01ac7f52>\", line 8, in scrapeJob\n",
      "    browser.get('https://www.tigerairtw.com/zh-tw')\n",
      "  File \"/home/james/.local/lib/python3.6/site-packages/selenium/webdriver/remote/webdriver.py\", line 333, in get\n",
      "    self.execute(Command.GET, {'url': url})\n",
      "  File \"/home/james/.local/lib/python3.6/site-packages/selenium/webdriver/remote/webdriver.py\", line 319, in execute\n",
      "    response = self.command_executor.execute(driver_command, params)\n",
      "  File \"/home/james/.local/lib/python3.6/site-packages/selenium/webdriver/remote/remote_connection.py\", line 374, in execute\n",
      "    return self._request(command_info[0], url, body=data)\n",
      "  File \"/home/james/.local/lib/python3.6/site-packages/selenium/webdriver/remote/remote_connection.py\", line 397, in _request\n",
      "    resp = self._conn.request(method, url, body=body, headers=headers)\n",
      "  File \"/home/james/.local/lib/python3.6/site-packages/urllib3/request.py\", line 72, in request\n",
      "    **urlopen_kw)\n",
      "  File \"/home/james/.local/lib/python3.6/site-packages/urllib3/request.py\", line 150, in request_encode_body\n",
      "    return self.urlopen(method, url, **extra_kw)\n",
      "  File \"/home/james/.local/lib/python3.6/site-packages/urllib3/poolmanager.py\", line 323, in urlopen\n",
      "    response = conn.urlopen(method, u.request_uri, **kw)\n",
      "  File \"/home/james/.local/lib/python3.6/site-packages/urllib3/connectionpool.py\", line 638, in urlopen\n",
      "    _stacktrace=sys.exc_info()[2])\n",
      "  File \"/home/james/.local/lib/python3.6/site-packages/urllib3/util/retry.py\", line 367, in increment\n",
      "    raise six.reraise(type(error), error, _stacktrace)\n",
      "  File \"/home/james/.local/lib/python3.6/site-packages/urllib3/packages/six.py\", line 685, in reraise\n",
      "    raise value.with_traceback(tb)\n",
      "  File \"/home/james/.local/lib/python3.6/site-packages/urllib3/connectionpool.py\", line 600, in urlopen\n",
      "    chunked=chunked)\n",
      "  File \"/home/james/.local/lib/python3.6/site-packages/urllib3/connectionpool.py\", line 384, in _make_request\n",
      "    six.raise_from(e, None)\n",
      "  File \"<string>\", line 2, in raise_from\n",
      "  File \"/home/james/.local/lib/python3.6/site-packages/urllib3/connectionpool.py\", line 380, in _make_request\n",
      "    httplib_response = conn.getresponse()\n",
      "  File \"/usr/lib/python3.6/http/client.py\", line 1331, in getresponse\n",
      "    response.begin()\n",
      "  File \"/usr/lib/python3.6/http/client.py\", line 297, in begin\n",
      "    version, status, reason = self._read_status()\n",
      "  File \"/usr/lib/python3.6/http/client.py\", line 266, in _read_status\n",
      "    raise RemoteDisconnected(\"Remote end closed connection without\"\n",
      "urllib3.exceptions.ProtocolError: ('Connection aborted.', RemoteDisconnected('Remote end closed connection without response',))\n",
      "\n",
      "Exception in thread Thread-294:\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/james/.local/lib/python3.6/site-packages/urllib3/connectionpool.py\", line 600, in urlopen\n",
      "    chunked=chunked)\n",
      "  File \"/home/james/.local/lib/python3.6/site-packages/urllib3/connectionpool.py\", line 384, in _make_request\n",
      "    six.raise_from(e, None)\n",
      "  File \"<string>\", line 2, in raise_from\n",
      "  File \"/home/james/.local/lib/python3.6/site-packages/urllib3/connectionpool.py\", line 380, in _make_request\n",
      "    httplib_response = conn.getresponse()\n",
      "  File \"/usr/lib/python3.6/http/client.py\", line 1331, in getresponse\n",
      "    response.begin()\n",
      "  File \"/usr/lib/python3.6/http/client.py\", line 297, in begin\n",
      "    version, status, reason = self._read_status()\n",
      "  File \"/usr/lib/python3.6/http/client.py\", line 266, in _read_status\n",
      "    raise RemoteDisconnected(\"Remote end closed connection without\"\n",
      "http.client.RemoteDisconnected: Remote end closed connection without response\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"/usr/lib/python3.6/threading.py\", line 916, in _bootstrap_inner\n",
      "    self.run()\n",
      "  File \"/usr/lib/python3.6/threading.py\", line 864, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"<ipython-input-49-a61b01ac7f52>\", line 8, in scrapeJob\n",
      "    browser.get('https://www.tigerairtw.com/zh-tw')\n",
      "  File \"/home/james/.local/lib/python3.6/site-packages/selenium/webdriver/remote/webdriver.py\", line 333, in get\n",
      "    self.execute(Command.GET, {'url': url})\n",
      "  File \"/home/james/.local/lib/python3.6/site-packages/selenium/webdriver/remote/webdriver.py\", line 319, in execute\n",
      "    response = self.command_executor.execute(driver_command, params)\n",
      "  File \"/home/james/.local/lib/python3.6/site-packages/selenium/webdriver/remote/remote_connection.py\", line 374, in execute\n",
      "    return self._request(command_info[0], url, body=data)\n",
      "  File \"/home/james/.local/lib/python3.6/site-packages/selenium/webdriver/remote/remote_connection.py\", line 397, in _request\n",
      "    resp = self._conn.request(method, url, body=body, headers=headers)\n",
      "  File \"/home/james/.local/lib/python3.6/site-packages/urllib3/request.py\", line 72, in request\n",
      "    **urlopen_kw)\n",
      "  File \"/home/james/.local/lib/python3.6/site-packages/urllib3/request.py\", line 150, in request_encode_body\n",
      "    return self.urlopen(method, url, **extra_kw)\n",
      "  File \"/home/james/.local/lib/python3.6/site-packages/urllib3/poolmanager.py\", line 323, in urlopen\n",
      "    response = conn.urlopen(method, u.request_uri, **kw)\n",
      "  File \"/home/james/.local/lib/python3.6/site-packages/urllib3/connectionpool.py\", line 638, in urlopen\n",
      "    _stacktrace=sys.exc_info()[2])\n",
      "  File \"/home/james/.local/lib/python3.6/site-packages/urllib3/util/retry.py\", line 367, in increment\n",
      "    raise six.reraise(type(error), error, _stacktrace)\n",
      "  File \"/home/james/.local/lib/python3.6/site-packages/urllib3/packages/six.py\", line 685, in reraise\n",
      "    raise value.with_traceback(tb)\n",
      "  File \"/home/james/.local/lib/python3.6/site-packages/urllib3/connectionpool.py\", line 600, in urlopen\n",
      "    chunked=chunked)\n",
      "  File \"/home/james/.local/lib/python3.6/site-packages/urllib3/connectionpool.py\", line 384, in _make_request\n",
      "    six.raise_from(e, None)\n",
      "  File \"<string>\", line 2, in raise_from\n",
      "  File \"/home/james/.local/lib/python3.6/site-packages/urllib3/connectionpool.py\", line 380, in _make_request\n",
      "    httplib_response = conn.getresponse()\n",
      "  File \"/usr/lib/python3.6/http/client.py\", line 1331, in getresponse\n",
      "    response.begin()\n",
      "  File \"/usr/lib/python3.6/http/client.py\", line 297, in begin\n",
      "    version, status, reason = self._read_status()\n",
      "  File \"/usr/lib/python3.6/http/client.py\", line 266, in _read_status\n",
      "    raise RemoteDisconnected(\"Remote end closed connection without\"\n",
      "urllib3.exceptions.ProtocolError: ('Connection aborted.', RemoteDisconnected('Remote end closed connection without response',))\n",
      "\n",
      "Exception in thread Thread-315:\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/james/.local/lib/python3.6/site-packages/urllib3/connectionpool.py\", line 600, in urlopen\n",
      "    chunked=chunked)\n",
      "  File \"/home/james/.local/lib/python3.6/site-packages/urllib3/connectionpool.py\", line 384, in _make_request\n",
      "    six.raise_from(e, None)\n",
      "  File \"<string>\", line 2, in raise_from\n",
      "  File \"/home/james/.local/lib/python3.6/site-packages/urllib3/connectionpool.py\", line 380, in _make_request\n",
      "    httplib_response = conn.getresponse()\n",
      "  File \"/usr/lib/python3.6/http/client.py\", line 1331, in getresponse\n",
      "    response.begin()\n",
      "  File \"/usr/lib/python3.6/http/client.py\", line 297, in begin\n",
      "    version, status, reason = self._read_status()\n",
      "  File \"/usr/lib/python3.6/http/client.py\", line 266, in _read_status\n",
      "    raise RemoteDisconnected(\"Remote end closed connection without\"\n",
      "http.client.RemoteDisconnected: Remote end closed connection without response\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"/usr/lib/python3.6/threading.py\", line 916, in _bootstrap_inner\n",
      "    self.run()\n",
      "  File \"/usr/lib/python3.6/threading.py\", line 864, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"<ipython-input-49-a61b01ac7f52>\", line 8, in scrapeJob\n",
      "    browser.get('https://www.tigerairtw.com/zh-tw')\n",
      "  File \"/home/james/.local/lib/python3.6/site-packages/selenium/webdriver/remote/webdriver.py\", line 333, in get\n",
      "    self.execute(Command.GET, {'url': url})\n",
      "  File \"/home/james/.local/lib/python3.6/site-packages/selenium/webdriver/remote/webdriver.py\", line 319, in execute\n",
      "    response = self.command_executor.execute(driver_command, params)\n",
      "  File \"/home/james/.local/lib/python3.6/site-packages/selenium/webdriver/remote/remote_connection.py\", line 374, in execute\n",
      "    return self._request(command_info[0], url, body=data)\n",
      "  File \"/home/james/.local/lib/python3.6/site-packages/selenium/webdriver/remote/remote_connection.py\", line 397, in _request\n",
      "    resp = self._conn.request(method, url, body=body, headers=headers)\n",
      "  File \"/home/james/.local/lib/python3.6/site-packages/urllib3/request.py\", line 72, in request\n",
      "    **urlopen_kw)\n",
      "  File \"/home/james/.local/lib/python3.6/site-packages/urllib3/request.py\", line 150, in request_encode_body\n",
      "    return self.urlopen(method, url, **extra_kw)\n",
      "  File \"/home/james/.local/lib/python3.6/site-packages/urllib3/poolmanager.py\", line 323, in urlopen\n",
      "    response = conn.urlopen(method, u.request_uri, **kw)\n",
      "  File \"/home/james/.local/lib/python3.6/site-packages/urllib3/connectionpool.py\", line 638, in urlopen\n",
      "    _stacktrace=sys.exc_info()[2])\n",
      "  File \"/home/james/.local/lib/python3.6/site-packages/urllib3/util/retry.py\", line 367, in increment\n",
      "    raise six.reraise(type(error), error, _stacktrace)\n",
      "  File \"/home/james/.local/lib/python3.6/site-packages/urllib3/packages/six.py\", line 685, in reraise\n",
      "    raise value.with_traceback(tb)\n",
      "  File \"/home/james/.local/lib/python3.6/site-packages/urllib3/connectionpool.py\", line 600, in urlopen\n",
      "    chunked=chunked)\n",
      "  File \"/home/james/.local/lib/python3.6/site-packages/urllib3/connectionpool.py\", line 384, in _make_request\n",
      "    six.raise_from(e, None)\n",
      "  File \"<string>\", line 2, in raise_from\n",
      "  File \"/home/james/.local/lib/python3.6/site-packages/urllib3/connectionpool.py\", line 380, in _make_request\n",
      "    httplib_response = conn.getresponse()\n",
      "  File \"/usr/lib/python3.6/http/client.py\", line 1331, in getresponse\n",
      "    response.begin()\n",
      "  File \"/usr/lib/python3.6/http/client.py\", line 297, in begin\n",
      "    version, status, reason = self._read_status()\n",
      "  File \"/usr/lib/python3.6/http/client.py\", line 266, in _read_status\n",
      "    raise RemoteDisconnected(\"Remote end closed connection without\"\n",
      "urllib3.exceptions.ProtocolError: ('Connection aborted.', RemoteDisconnected('Remote end closed connection without response',))\n",
      "\n",
      "Exception in thread Thread-314:\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/james/.local/lib/python3.6/site-packages/urllib3/connectionpool.py\", line 600, in urlopen\n",
      "    chunked=chunked)\n",
      "  File \"/home/james/.local/lib/python3.6/site-packages/urllib3/connectionpool.py\", line 384, in _make_request\n",
      "    six.raise_from(e, None)\n",
      "  File \"<string>\", line 2, in raise_from\n",
      "  File \"/home/james/.local/lib/python3.6/site-packages/urllib3/connectionpool.py\", line 380, in _make_request\n",
      "    httplib_response = conn.getresponse()\n",
      "  File \"/usr/lib/python3.6/http/client.py\", line 1331, in getresponse\n",
      "    response.begin()\n",
      "  File \"/usr/lib/python3.6/http/client.py\", line 297, in begin\n",
      "    version, status, reason = self._read_status()\n",
      "  File \"/usr/lib/python3.6/http/client.py\", line 266, in _read_status\n",
      "    raise RemoteDisconnected(\"Remote end closed connection without\"\n",
      "http.client.RemoteDisconnected: Remote end closed connection without response\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"/usr/lib/python3.6/threading.py\", line 916, in _bootstrap_inner\n",
      "    self.run()\n",
      "  File \"/usr/lib/python3.6/threading.py\", line 864, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"<ipython-input-49-a61b01ac7f52>\", line 8, in scrapeJob\n",
      "    browser.get('https://www.tigerairtw.com/zh-tw')\n",
      "  File \"/home/james/.local/lib/python3.6/site-packages/selenium/webdriver/remote/webdriver.py\", line 333, in get\n",
      "    self.execute(Command.GET, {'url': url})\n",
      "  File \"/home/james/.local/lib/python3.6/site-packages/selenium/webdriver/remote/webdriver.py\", line 319, in execute\n",
      "    response = self.command_executor.execute(driver_command, params)\n",
      "  File \"/home/james/.local/lib/python3.6/site-packages/selenium/webdriver/remote/remote_connection.py\", line 374, in execute\n",
      "    return self._request(command_info[0], url, body=data)\n",
      "  File \"/home/james/.local/lib/python3.6/site-packages/selenium/webdriver/remote/remote_connection.py\", line 397, in _request\n",
      "    resp = self._conn.request(method, url, body=body, headers=headers)\n",
      "  File \"/home/james/.local/lib/python3.6/site-packages/urllib3/request.py\", line 72, in request\n",
      "    **urlopen_kw)\n",
      "  File \"/home/james/.local/lib/python3.6/site-packages/urllib3/request.py\", line 150, in request_encode_body\n",
      "    return self.urlopen(method, url, **extra_kw)\n",
      "  File \"/home/james/.local/lib/python3.6/site-packages/urllib3/poolmanager.py\", line 323, in urlopen\n",
      "    response = conn.urlopen(method, u.request_uri, **kw)\n",
      "  File \"/home/james/.local/lib/python3.6/site-packages/urllib3/connectionpool.py\", line 638, in urlopen\n",
      "    _stacktrace=sys.exc_info()[2])\n",
      "  File \"/home/james/.local/lib/python3.6/site-packages/urllib3/util/retry.py\", line 367, in increment\n",
      "    raise six.reraise(type(error), error, _stacktrace)\n",
      "  File \"/home/james/.local/lib/python3.6/site-packages/urllib3/packages/six.py\", line 685, in reraise\n",
      "    raise value.with_traceback(tb)\n",
      "  File \"/home/james/.local/lib/python3.6/site-packages/urllib3/connectionpool.py\", line 600, in urlopen\n",
      "    chunked=chunked)\n",
      "  File \"/home/james/.local/lib/python3.6/site-packages/urllib3/connectionpool.py\", line 384, in _make_request\n",
      "    six.raise_from(e, None)\n",
      "  File \"<string>\", line 2, in raise_from\n",
      "  File \"/home/james/.local/lib/python3.6/site-packages/urllib3/connectionpool.py\", line 380, in _make_request\n",
      "    httplib_response = conn.getresponse()\n",
      "  File \"/usr/lib/python3.6/http/client.py\", line 1331, in getresponse\n",
      "    response.begin()\n",
      "  File \"/usr/lib/python3.6/http/client.py\", line 297, in begin\n",
      "    version, status, reason = self._read_status()\n",
      "  File \"/usr/lib/python3.6/http/client.py\", line 266, in _read_status\n",
      "    raise RemoteDisconnected(\"Remote end closed connection without\"\n",
      "urllib3.exceptions.ProtocolError: ('Connection aborted.', RemoteDisconnected('Remote end closed connection without response',))\n",
      "\n",
      "Exception in thread Thread-304:\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/james/.local/lib/python3.6/site-packages/urllib3/connectionpool.py\", line 600, in urlopen\n",
      "    chunked=chunked)\n",
      "  File \"/home/james/.local/lib/python3.6/site-packages/urllib3/connectionpool.py\", line 384, in _make_request\n",
      "    six.raise_from(e, None)\n",
      "  File \"<string>\", line 2, in raise_from\n",
      "  File \"/home/james/.local/lib/python3.6/site-packages/urllib3/connectionpool.py\", line 380, in _make_request\n",
      "    httplib_response = conn.getresponse()\n",
      "  File \"/usr/lib/python3.6/http/client.py\", line 1331, in getresponse\n",
      "    response.begin()\n",
      "  File \"/usr/lib/python3.6/http/client.py\", line 297, in begin\n",
      "    version, status, reason = self._read_status()\n",
      "  File \"/usr/lib/python3.6/http/client.py\", line 266, in _read_status\n",
      "    raise RemoteDisconnected(\"Remote end closed connection without\"\n",
      "http.client.RemoteDisconnected: Remote end closed connection without response\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"/usr/lib/python3.6/threading.py\", line 916, in _bootstrap_inner\n",
      "    self.run()\n",
      "  File \"/usr/lib/python3.6/threading.py\", line 864, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"<ipython-input-49-a61b01ac7f52>\", line 8, in scrapeJob\n",
      "    browser.get('https://www.tigerairtw.com/zh-tw')\n",
      "  File \"/home/james/.local/lib/python3.6/site-packages/selenium/webdriver/remote/webdriver.py\", line 333, in get\n",
      "    self.execute(Command.GET, {'url': url})\n",
      "  File \"/home/james/.local/lib/python3.6/site-packages/selenium/webdriver/remote/webdriver.py\", line 319, in execute\n",
      "    response = self.command_executor.execute(driver_command, params)\n",
      "  File \"/home/james/.local/lib/python3.6/site-packages/selenium/webdriver/remote/remote_connection.py\", line 374, in execute\n",
      "    return self._request(command_info[0], url, body=data)\n",
      "  File \"/home/james/.local/lib/python3.6/site-packages/selenium/webdriver/remote/remote_connection.py\", line 397, in _request\n",
      "    resp = self._conn.request(method, url, body=body, headers=headers)\n",
      "  File \"/home/james/.local/lib/python3.6/site-packages/urllib3/request.py\", line 72, in request\n",
      "    **urlopen_kw)\n",
      "  File \"/home/james/.local/lib/python3.6/site-packages/urllib3/request.py\", line 150, in request_encode_body\n",
      "    return self.urlopen(method, url, **extra_kw)\n",
      "  File \"/home/james/.local/lib/python3.6/site-packages/urllib3/poolmanager.py\", line 323, in urlopen\n",
      "    response = conn.urlopen(method, u.request_uri, **kw)\n",
      "  File \"/home/james/.local/lib/python3.6/site-packages/urllib3/connectionpool.py\", line 638, in urlopen\n",
      "    _stacktrace=sys.exc_info()[2])\n",
      "  File \"/home/james/.local/lib/python3.6/site-packages/urllib3/util/retry.py\", line 367, in increment\n",
      "    raise six.reraise(type(error), error, _stacktrace)\n",
      "  File \"/home/james/.local/lib/python3.6/site-packages/urllib3/packages/six.py\", line 685, in reraise\n",
      "    raise value.with_traceback(tb)\n",
      "  File \"/home/james/.local/lib/python3.6/site-packages/urllib3/connectionpool.py\", line 600, in urlopen\n",
      "    chunked=chunked)\n",
      "  File \"/home/james/.local/lib/python3.6/site-packages/urllib3/connectionpool.py\", line 384, in _make_request\n",
      "    six.raise_from(e, None)\n",
      "  File \"<string>\", line 2, in raise_from\n",
      "  File \"/home/james/.local/lib/python3.6/site-packages/urllib3/connectionpool.py\", line 380, in _make_request\n",
      "    httplib_response = conn.getresponse()\n",
      "  File \"/usr/lib/python3.6/http/client.py\", line 1331, in getresponse\n",
      "    response.begin()\n",
      "  File \"/usr/lib/python3.6/http/client.py\", line 297, in begin\n",
      "    version, status, reason = self._read_status()\n",
      "  File \"/usr/lib/python3.6/http/client.py\", line 266, in _read_status\n",
      "    raise RemoteDisconnected(\"Remote end closed connection without\"\n",
      "urllib3.exceptions.ProtocolError: ('Connection aborted.', RemoteDisconnected('Remote end closed connection without response',))\n",
      "\n",
      "Exception in thread Thread-317:\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/james/.local/lib/python3.6/site-packages/urllib3/connectionpool.py\", line 600, in urlopen\n",
      "    chunked=chunked)\n",
      "  File \"/home/james/.local/lib/python3.6/site-packages/urllib3/connectionpool.py\", line 384, in _make_request\n",
      "    six.raise_from(e, None)\n",
      "  File \"<string>\", line 2, in raise_from\n",
      "  File \"/home/james/.local/lib/python3.6/site-packages/urllib3/connectionpool.py\", line 380, in _make_request\n",
      "    httplib_response = conn.getresponse()\n",
      "  File \"/usr/lib/python3.6/http/client.py\", line 1331, in getresponse\n",
      "    response.begin()\n",
      "  File \"/usr/lib/python3.6/http/client.py\", line 297, in begin\n",
      "    version, status, reason = self._read_status()\n",
      "  File \"/usr/lib/python3.6/http/client.py\", line 266, in _read_status\n",
      "    raise RemoteDisconnected(\"Remote end closed connection without\"\n",
      "http.client.RemoteDisconnected: Remote end closed connection without response\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"/usr/lib/python3.6/threading.py\", line 916, in _bootstrap_inner\n",
      "    self.run()\n",
      "  File \"/usr/lib/python3.6/threading.py\", line 864, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"<ipython-input-49-a61b01ac7f52>\", line 7, in scrapeJob\n",
      "    browser = webdriver.Chrome(executable_path='driver/chromedriver')\n",
      "  File \"/home/james/.local/lib/python3.6/site-packages/selenium/webdriver/chrome/webdriver.py\", line 81, in __init__\n",
      "    desired_capabilities=desired_capabilities)\n",
      "  File \"/home/james/.local/lib/python3.6/site-packages/selenium/webdriver/remote/webdriver.py\", line 157, in __init__\n",
      "    self.start_session(capabilities, browser_profile)\n",
      "  File \"/home/james/.local/lib/python3.6/site-packages/selenium/webdriver/remote/webdriver.py\", line 252, in start_session\n",
      "    response = self.execute(Command.NEW_SESSION, parameters)\n",
      "  File \"/home/james/.local/lib/python3.6/site-packages/selenium/webdriver/remote/webdriver.py\", line 319, in execute\n",
      "    response = self.command_executor.execute(driver_command, params)\n",
      "  File \"/home/james/.local/lib/python3.6/site-packages/selenium/webdriver/remote/remote_connection.py\", line 374, in execute\n",
      "    return self._request(command_info[0], url, body=data)\n",
      "  File \"/home/james/.local/lib/python3.6/site-packages/selenium/webdriver/remote/remote_connection.py\", line 397, in _request\n",
      "    resp = self._conn.request(method, url, body=body, headers=headers)\n",
      "  File \"/home/james/.local/lib/python3.6/site-packages/urllib3/request.py\", line 72, in request\n",
      "    **urlopen_kw)\n",
      "  File \"/home/james/.local/lib/python3.6/site-packages/urllib3/request.py\", line 150, in request_encode_body\n",
      "    return self.urlopen(method, url, **extra_kw)\n",
      "  File \"/home/james/.local/lib/python3.6/site-packages/urllib3/poolmanager.py\", line 323, in urlopen\n",
      "    response = conn.urlopen(method, u.request_uri, **kw)\n",
      "  File \"/home/james/.local/lib/python3.6/site-packages/urllib3/connectionpool.py\", line 638, in urlopen\n",
      "    _stacktrace=sys.exc_info()[2])\n",
      "  File \"/home/james/.local/lib/python3.6/site-packages/urllib3/util/retry.py\", line 367, in increment\n",
      "    raise six.reraise(type(error), error, _stacktrace)\n",
      "  File \"/home/james/.local/lib/python3.6/site-packages/urllib3/packages/six.py\", line 685, in reraise\n",
      "    raise value.with_traceback(tb)\n",
      "  File \"/home/james/.local/lib/python3.6/site-packages/urllib3/connectionpool.py\", line 600, in urlopen\n",
      "    chunked=chunked)\n",
      "  File \"/home/james/.local/lib/python3.6/site-packages/urllib3/connectionpool.py\", line 384, in _make_request\n",
      "    six.raise_from(e, None)\n",
      "  File \"<string>\", line 2, in raise_from\n",
      "  File \"/home/james/.local/lib/python3.6/site-packages/urllib3/connectionpool.py\", line 380, in _make_request\n",
      "    httplib_response = conn.getresponse()\n",
      "  File \"/usr/lib/python3.6/http/client.py\", line 1331, in getresponse\n",
      "    response.begin()\n",
      "  File \"/usr/lib/python3.6/http/client.py\", line 297, in begin\n",
      "    version, status, reason = self._read_status()\n",
      "  File \"/usr/lib/python3.6/http/client.py\", line 266, in _read_status\n",
      "    raise RemoteDisconnected(\"Remote end closed connection without\"\n",
      "urllib3.exceptions.ProtocolError: ('Connection aborted.', RemoteDisconnected('Remote end closed connection without response',))\n",
      "\n",
      "Exception in thread Thread-319:\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/james/.local/lib/python3.6/site-packages/urllib3/connectionpool.py\", line 600, in urlopen\n",
      "    chunked=chunked)\n",
      "  File \"/home/james/.local/lib/python3.6/site-packages/urllib3/connectionpool.py\", line 384, in _make_request\n",
      "    six.raise_from(e, None)\n",
      "  File \"<string>\", line 2, in raise_from\n",
      "  File \"/home/james/.local/lib/python3.6/site-packages/urllib3/connectionpool.py\", line 380, in _make_request\n",
      "    httplib_response = conn.getresponse()\n",
      "  File \"/usr/lib/python3.6/http/client.py\", line 1331, in getresponse\n",
      "    response.begin()\n",
      "  File \"/usr/lib/python3.6/http/client.py\", line 297, in begin\n",
      "    version, status, reason = self._read_status()\n",
      "  File \"/usr/lib/python3.6/http/client.py\", line 266, in _read_status\n",
      "    raise RemoteDisconnected(\"Remote end closed connection without\"\n",
      "http.client.RemoteDisconnected: Remote end closed connection without response\n",
      "\n",
      "During handling of the above exception, another exception occurred:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"/usr/lib/python3.6/threading.py\", line 916, in _bootstrap_inner\n",
      "    self.run()\n",
      "  File \"/usr/lib/python3.6/threading.py\", line 864, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"<ipython-input-49-a61b01ac7f52>\", line 7, in scrapeJob\n",
      "    browser = webdriver.Chrome(executable_path='driver/chromedriver')\n",
      "  File \"/home/james/.local/lib/python3.6/site-packages/selenium/webdriver/chrome/webdriver.py\", line 81, in __init__\n",
      "    desired_capabilities=desired_capabilities)\n",
      "  File \"/home/james/.local/lib/python3.6/site-packages/selenium/webdriver/remote/webdriver.py\", line 157, in __init__\n",
      "    self.start_session(capabilities, browser_profile)\n",
      "  File \"/home/james/.local/lib/python3.6/site-packages/selenium/webdriver/remote/webdriver.py\", line 252, in start_session\n",
      "    response = self.execute(Command.NEW_SESSION, parameters)\n",
      "  File \"/home/james/.local/lib/python3.6/site-packages/selenium/webdriver/remote/webdriver.py\", line 319, in execute\n",
      "    response = self.command_executor.execute(driver_command, params)\n",
      "  File \"/home/james/.local/lib/python3.6/site-packages/selenium/webdriver/remote/remote_connection.py\", line 374, in execute\n",
      "    return self._request(command_info[0], url, body=data)\n",
      "  File \"/home/james/.local/lib/python3.6/site-packages/selenium/webdriver/remote/remote_connection.py\", line 397, in _request\n",
      "    resp = self._conn.request(method, url, body=body, headers=headers)\n",
      "  File \"/home/james/.local/lib/python3.6/site-packages/urllib3/request.py\", line 72, in request\n",
      "    **urlopen_kw)\n",
      "  File \"/home/james/.local/lib/python3.6/site-packages/urllib3/request.py\", line 150, in request_encode_body\n",
      "    return self.urlopen(method, url, **extra_kw)\n",
      "  File \"/home/james/.local/lib/python3.6/site-packages/urllib3/poolmanager.py\", line 323, in urlopen\n",
      "    response = conn.urlopen(method, u.request_uri, **kw)\n",
      "  File \"/home/james/.local/lib/python3.6/site-packages/urllib3/connectionpool.py\", line 638, in urlopen\n",
      "    _stacktrace=sys.exc_info()[2])\n",
      "  File \"/home/james/.local/lib/python3.6/site-packages/urllib3/util/retry.py\", line 367, in increment\n",
      "    raise six.reraise(type(error), error, _stacktrace)\n",
      "  File \"/home/james/.local/lib/python3.6/site-packages/urllib3/packages/six.py\", line 685, in reraise\n",
      "    raise value.with_traceback(tb)\n",
      "  File \"/home/james/.local/lib/python3.6/site-packages/urllib3/connectionpool.py\", line 600, in urlopen\n",
      "    chunked=chunked)\n",
      "  File \"/home/james/.local/lib/python3.6/site-packages/urllib3/connectionpool.py\", line 384, in _make_request\n",
      "    six.raise_from(e, None)\n",
      "  File \"<string>\", line 2, in raise_from\n",
      "  File \"/home/james/.local/lib/python3.6/site-packages/urllib3/connectionpool.py\", line 380, in _make_request\n",
      "    httplib_response = conn.getresponse()\n",
      "  File \"/usr/lib/python3.6/http/client.py\", line 1331, in getresponse\n",
      "    response.begin()\n",
      "  File \"/usr/lib/python3.6/http/client.py\", line 297, in begin\n",
      "    version, status, reason = self._read_status()\n",
      "  File \"/usr/lib/python3.6/http/client.py\", line 266, in _read_status\n",
      "    raise RemoteDisconnected(\"Remote end closed connection without\"\n",
      "urllib3.exceptions.ProtocolError: ('Connection aborted.', RemoteDisconnected('Remote end closed connection without response',))\n",
      "\n"
     ]
    },
    {
     "ename": "KeyboardInterrupt",
     "evalue": "",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mKeyboardInterrupt\u001b[0m                         Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-52-705a48aed2e7>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[1;32m      4\u001b[0m     \u001b[0mthreads\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0mi\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mstart\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      5\u001b[0m     \u001b[0;32mif\u001b[0m \u001b[0mi\u001b[0m \u001b[0;34m%\u001b[0m  \u001b[0;36m3\u001b[0m \u001b[0;34m==\u001b[0m \u001b[0;36m0\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m----> 6\u001b[0;31m         \u001b[0mtime\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0msleep\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;36m8\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m      7\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      8\u001b[0m \u001b[0;32mfor\u001b[0m \u001b[0mi\u001b[0m \u001b[0;32min\u001b[0m \u001b[0mrange\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;36m300\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mKeyboardInterrupt\u001b[0m: "
     ]
    }
   ],
   "source": [
    "threads = []\n",
    "for i in range(300):\n",
    "    threads.append(threading.Thread(target = scrapeJob, args = (i+72,0,'XX3','XXX')))\n",
    "    threads[i].start()\n",
    "    if i %  3 == 0:\n",
    "        time.sleep(11)\n",
    "\n",
    "for i in range(300):\n",
    "    print(i, ' thread has finished')\n",
    "    threads[i].join()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.7"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
